// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/AppleMessageOverridePayload.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(a, n) { if (!(a instanceof n)) throw new TypeError("Cannot call a class as a function"); }
function _defineProperties(e, r) { for (var t = 0; t < r.length; t++) { var o = r[t]; o.enumerable = o.enumerable || !1, o.configurable = !0, "value" in o && (o.writable = !0), Object.defineProperty(e, _toPropertyKey(o.key), o); } }
function _createClass(e, r, t) { return r && _defineProperties(e.prototype, r), t && _defineProperties(e, t), Object.defineProperty(e, "prototype", { writable: !1 }), e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Sunshine Conversations API
 *
 * The version of the OpenAPI document: 16.0.1
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 *
 * OpenAPI Generator version: 4.3.1
 *
 * Do not edit the class manually.
 *
 */
/**
 * The AppleMessageOverridePayload model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/AppleMessageOverridePayload
 * @version 16.0.1
 */
var AppleMessageOverridePayload = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AppleMessageOverridePayload</code>.
   * The exact payload to send to the channel.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/AppleMessageOverridePayload
   */
  function AppleMessageOverridePayload() {
    _classCallCheck(this, AppleMessageOverridePayload);
    AppleMessageOverridePayload.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  return _createClass(AppleMessageOverridePayload, [{
    key: "getPayload",
    value:
    /**
         * @return {Object}
         */
    function getPayload() {
      return this.payload;
    }

    /**
     * @param {Object} payload
     */
  }, {
    key: "setPayload",
    value: function setPayload(payload) {
      this['payload'] = payload;
    }
    /**
         * Returns List of capabilities needed for the override message to be received by the end user.   `LIST` : for list picker; `TIME` : for time picker; `FORM` : for form; `QUICK` : for quick reply; `AUTH2` : for authentication 
         * @return {Array.<module:sunshine-conversations-client/sunshine-conversations-client.model/AppleMessageOverridePayload.WithCapabilitiesEnum>}
         */
  }, {
    key: "getWithCapabilities",
    value: function getWithCapabilities() {
      return this.withCapabilities;
    }

    /**
     * Sets List of capabilities needed for the override message to be received by the end user.   `LIST` : for list picker; `TIME` : for time picker; `FORM` : for form; `QUICK` : for quick reply; `AUTH2` : for authentication 
     * @param {Array.<module:sunshine-conversations-client/sunshine-conversations-client.model/AppleMessageOverridePayload.WithCapabilitiesEnum>} withCapabilities List of capabilities needed for the override message to be received by the end user.   `LIST` : for list picker; `TIME` : for time picker; `FORM` : for form; `QUICK` : for quick reply; `AUTH2` : for authentication 
     */
  }, {
    key: "setWithCapabilities",
    value: function setWithCapabilities(withCapabilities) {
      this['withCapabilities'] = withCapabilities;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>AppleMessageOverridePayload</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/AppleMessageOverridePayload} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/AppleMessageOverridePayload} The populated <code>AppleMessageOverridePayload</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AppleMessageOverridePayload();
        if (data.hasOwnProperty('payload')) {
          obj['payload'] = _ApiClient["default"].convertToType(data['payload'], Object);
        }
        if (data.hasOwnProperty('withCapabilities')) {
          obj['withCapabilities'] = _ApiClient["default"].convertToType(data['withCapabilities'], ['String']);
        }
      }
      return obj;
    }
  }]);
}();
/**
 * @member {Object} payload
 */
AppleMessageOverridePayload.prototype['payload'] = undefined;

/**
 * List of capabilities needed for the override message to be received by the end user.   `LIST` : for list picker; `TIME` : for time picker; `FORM` : for form; `QUICK` : for quick reply; `AUTH2` : for authentication 
 * @member {Array.<module:sunshine-conversations-client/sunshine-conversations-client.model/AppleMessageOverridePayload.WithCapabilitiesEnum>} withCapabilities
 */
AppleMessageOverridePayload.prototype['withCapabilities'] = undefined;

/**
 * Allowed values for the <code>withCapabilities</code> property.
 * @enum {String}
 * @readonly
 */
AppleMessageOverridePayload['WithCapabilitiesEnum'] = {
  /**
   * value: "LIST"
   * @const
   */
  "LIST": "LIST",
  /**
   * value: "TIME"
   * @const
   */
  "TIME": "TIME",
  /**
   * value: "FORM"
   * @const
   */
  "FORM": "FORM",
  /**
   * value: "QUICK"
   * @const
   */
  "QUICK": "QUICK",
  /**
   * value: "AUTH2"
   * @const
   */
  "AUTH2": "AUTH2"
};
var _default = exports["default"] = AppleMessageOverridePayload;
