// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/ConversionEventsBody.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _InstagramConversionEventsBody = _interopRequireDefault(require("./InstagramConversionEventsBody"));
var _MessengerConversionEventsBody = _interopRequireDefault(require("./MessengerConversionEventsBody"));
var _MetaConversionEvent = _interopRequireDefault(require("./MetaConversionEvent"));
var _WhatsAppConversionEventsBody = _interopRequireDefault(require("./WhatsAppConversionEventsBody"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(a, n) { if (!(a instanceof n)) throw new TypeError("Cannot call a class as a function"); }
function _defineProperties(e, r) { for (var t = 0; t < r.length; t++) { var o = r[t]; o.enumerable = o.enumerable || !1, o.configurable = !0, "value" in o && (o.writable = !0), Object.defineProperty(e, _toPropertyKey(o.key), o); } }
function _createClass(e, r, t) { return r && _defineProperties(e.prototype, r), t && _defineProperties(e, t), Object.defineProperty(e, "prototype", { writable: !1 }), e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Sunshine Conversations API
 *
 * The version of the OpenAPI document: 16.0.1
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 *
 * OpenAPI Generator version: 4.3.1
 *
 * Do not edit the class manually.
 *
 */
/**
 * The ConversionEventsBody model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/ConversionEventsBody
 * @version 16.0.1
 */
var ConversionEventsBody = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ConversionEventsBody</code>.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/ConversionEventsBody
   * @implements module:sunshine-conversations-client/sunshine-conversations-client.model/MessengerConversionEventsBody
   * @implements module:sunshine-conversations-client/sunshine-conversations-client.model/InstagramConversionEventsBody
   * @implements module:sunshine-conversations-client/sunshine-conversations-client.model/WhatsAppConversionEventsBody
   */
  function ConversionEventsBody() {
    _classCallCheck(this, ConversionEventsBody);
    _MessengerConversionEventsBody["default"].initialize(this);
    _InstagramConversionEventsBody["default"].initialize(this);
    _WhatsAppConversionEventsBody["default"].initialize(this);
    ConversionEventsBody.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  return _createClass(ConversionEventsBody, [{
    key: "getMessenger",
    value:
    /**
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/MetaConversionEvent}
         */
    function getMessenger() {
      return this.messenger;
    }

    /**
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/MetaConversionEvent} messenger
     */
  }, {
    key: "setMessenger",
    value: function setMessenger(messenger) {
      this['messenger'] = messenger;
    }
    /**
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/MetaConversionEvent}
         */
  }, {
    key: "getInstagram",
    value: function getInstagram() {
      return this.instagram;
    }

    /**
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/MetaConversionEvent} instagram
     */
  }, {
    key: "setInstagram",
    value: function setInstagram(instagram) {
      this['instagram'] = instagram;
    }
    /**
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/MetaConversionEvent}
         */
  }, {
    key: "getWhatsapp",
    value: function getWhatsapp() {
      return this.whatsapp;
    }

    /**
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/MetaConversionEvent} whatsapp
     */
  }, {
    key: "setWhatsapp",
    value: function setWhatsapp(whatsapp) {
      this['whatsapp'] = whatsapp;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>ConversionEventsBody</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/ConversionEventsBody} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/ConversionEventsBody} The populated <code>ConversionEventsBody</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ConversionEventsBody();
        _MessengerConversionEventsBody["default"].constructFromObject(data, obj);
        _InstagramConversionEventsBody["default"].constructFromObject(data, obj);
        _WhatsAppConversionEventsBody["default"].constructFromObject(data, obj);
        if (data.hasOwnProperty('messenger')) {
          obj['messenger'] = _MetaConversionEvent["default"].constructFromObject(data['messenger']);
        }
        if (data.hasOwnProperty('instagram')) {
          obj['instagram'] = _MetaConversionEvent["default"].constructFromObject(data['instagram']);
        }
        if (data.hasOwnProperty('whatsapp')) {
          obj['whatsapp'] = _MetaConversionEvent["default"].constructFromObject(data['whatsapp']);
        }
      }
      return obj;
    }
  }]);
}();
/**
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/MetaConversionEvent} messenger
 */
ConversionEventsBody.prototype['messenger'] = undefined;

/**
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/MetaConversionEvent} instagram
 */
ConversionEventsBody.prototype['instagram'] = undefined;

/**
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/MetaConversionEvent} whatsapp
 */
ConversionEventsBody.prototype['whatsapp'] = undefined;

// Implement MessengerConversionEventsBody interface:
/**
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/MetaConversionEvent} messenger
 */
_MessengerConversionEventsBody["default"].prototype['messenger'] = undefined;
// Implement InstagramConversionEventsBody interface:
/**
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/MetaConversionEvent} instagram
 */
_InstagramConversionEventsBody["default"].prototype['instagram'] = undefined;
// Implement WhatsAppConversionEventsBody interface:
/**
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/MetaConversionEvent} whatsapp
 */
_WhatsAppConversionEventsBody["default"].prototype['whatsapp'] = undefined;
var _default = exports["default"] = ConversionEventsBody;
