// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/DefaultResponderDefaultResponder.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(a, n) { if (!(a instanceof n)) throw new TypeError("Cannot call a class as a function"); }
function _defineProperties(e, r) { for (var t = 0; t < r.length; t++) { var o = r[t]; o.enumerable = o.enumerable || !1, o.configurable = !0, "value" in o && (o.writable = !0), Object.defineProperty(e, _toPropertyKey(o.key), o); } }
function _createClass(e, r, t) { return r && _defineProperties(e.prototype, r), t && _defineProperties(e, t), Object.defineProperty(e, "prototype", { writable: !1 }), e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Sunshine Conversations API
 *
 * The version of the OpenAPI document: 16.0.1
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 *
 * OpenAPI Generator version: 4.3.1
 *
 * Do not edit the class manually.
 *
 */
/**
 * The DefaultResponderDefaultResponder model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/DefaultResponderDefaultResponder
 * @version 16.0.1
 */
var DefaultResponderDefaultResponder = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>DefaultResponderDefaultResponder</code>.
   * The default responder for the integration. This is the responder that will be used to send messages to the user. For more information, refer to the &lt;a href&#x3D;\&quot;https://developer.zendesk.com/documentation/conversations/messaging-platform/programmable-conversations/switchboard/#default-integration-assignment\&quot;&gt;Switchboard guide&lt;/a&gt;. 
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/DefaultResponderDefaultResponder
   */
  function DefaultResponderDefaultResponder() {
    _classCallCheck(this, DefaultResponderDefaultResponder);
    DefaultResponderDefaultResponder.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  return _createClass(DefaultResponderDefaultResponder, [{
    key: "getName",
    value:
    /**
         * Returns The name of the switchboard integration.
         * @return {String}
         */
    function getName() {
      return this.name;
    }

    /**
     * Sets The name of the switchboard integration.
     * @param {String} name The name of the switchboard integration.
     */
  }, {
    key: "setName",
    value: function setName(name) {
      this['name'] = name;
    }
    /**
         * Returns The unique ID of the integration.
         * @return {String}
         */
  }, {
    key: "getIntegrationId",
    value: function getIntegrationId() {
      return this.integrationId;
    }

    /**
     * Sets The unique ID of the integration.
     * @param {String} integrationId The unique ID of the integration.
     */
  }, {
    key: "setIntegrationId",
    value: function setIntegrationId(integrationId) {
      this['integrationId'] = integrationId;
    }
    /**
         * Returns The type of the integration.
         * @return {String}
         */
  }, {
    key: "getIntegrationType",
    value: function getIntegrationType() {
      return this.integrationType;
    }

    /**
     * Sets The type of the integration.
     * @param {String} integrationType The type of the integration.
     */
  }, {
    key: "setIntegrationType",
    value: function setIntegrationType(integrationType) {
      this['integrationType'] = integrationType;
    }
    /**
         * Returns Indicates whether the switchboard should deliver standby events.
         * @return {Boolean}
         */
  }, {
    key: "getDeliverStandbyEvents",
    value: function getDeliverStandbyEvents() {
      return this.deliverStandbyEvents;
    }

    /**
     * Sets Indicates whether the switchboard should deliver standby events.
     * @param {Boolean} deliverStandbyEvents Indicates whether the switchboard should deliver standby events.
     */
  }, {
    key: "setDeliverStandbyEvents",
    value: function setDeliverStandbyEvents(deliverStandbyEvents) {
      this['deliverStandbyEvents'] = deliverStandbyEvents;
    }
    /**
         * Returns The unique ID of the next switchboard integration.
         * @return {String}
         */
  }, {
    key: "getNextSwitchboardIntegrationId",
    value: function getNextSwitchboardIntegrationId() {
      return this.nextSwitchboardIntegrationId;
    }

    /**
     * Sets The unique ID of the next switchboard integration.
     * @param {String} nextSwitchboardIntegrationId The unique ID of the next switchboard integration.
     */
  }, {
    key: "setNextSwitchboardIntegrationId",
    value: function setNextSwitchboardIntegrationId(nextSwitchboardIntegrationId) {
      this['nextSwitchboardIntegrationId'] = nextSwitchboardIntegrationId;
    }
    /**
         * Returns The number of messages to keep in the message history.
         * @return {Number}
         */
  }, {
    key: "getMessageHistoryCount",
    value: function getMessageHistoryCount() {
      return this.messageHistoryCount;
    }

    /**
     * Sets The number of messages to keep in the message history.
     * @param {Number} messageHistoryCount The number of messages to keep in the message history.
     */
  }, {
    key: "setMessageHistoryCount",
    value: function setMessageHistoryCount(messageHistoryCount) {
      this['messageHistoryCount'] = messageHistoryCount;
    }
    /**
         * Returns Indicates whether the default responder is inherited from the switchboard's global config or not. Returns `false` if a per-channel responder override has been set for this integration, and `true` otherwise.
         * @return {Boolean}
         */
  }, {
    key: "getInherited",
    value: function getInherited() {
      return this.inherited;
    }

    /**
     * Sets Indicates whether the default responder is inherited from the switchboard's global config or not. Returns `false` if a per-channel responder override has been set for this integration, and `true` otherwise.
     * @param {Boolean} inherited Indicates whether the default responder is inherited from the switchboard's global config or not. Returns `false` if a per-channel responder override has been set for this integration, and `true` otherwise.
     */
  }, {
    key: "setInherited",
    value: function setInherited(inherited) {
      this['inherited'] = inherited;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>DefaultResponderDefaultResponder</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/DefaultResponderDefaultResponder} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/DefaultResponderDefaultResponder} The populated <code>DefaultResponderDefaultResponder</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new DefaultResponderDefaultResponder();
        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }
        if (data.hasOwnProperty('integrationId')) {
          obj['integrationId'] = _ApiClient["default"].convertToType(data['integrationId'], 'String');
        }
        if (data.hasOwnProperty('integrationType')) {
          obj['integrationType'] = _ApiClient["default"].convertToType(data['integrationType'], 'String');
        }
        if (data.hasOwnProperty('deliverStandbyEvents')) {
          obj['deliverStandbyEvents'] = _ApiClient["default"].convertToType(data['deliverStandbyEvents'], 'Boolean');
        }
        if (data.hasOwnProperty('nextSwitchboardIntegrationId')) {
          obj['nextSwitchboardIntegrationId'] = _ApiClient["default"].convertToType(data['nextSwitchboardIntegrationId'], 'String');
        }
        if (data.hasOwnProperty('messageHistoryCount')) {
          obj['messageHistoryCount'] = _ApiClient["default"].convertToType(data['messageHistoryCount'], 'Number');
        }
        if (data.hasOwnProperty('inherited')) {
          obj['inherited'] = _ApiClient["default"].convertToType(data['inherited'], 'Boolean');
        }
      }
      return obj;
    }
  }]);
}();
/**
 * The name of the switchboard integration.
 * @member {String} name
 */
DefaultResponderDefaultResponder.prototype['name'] = undefined;

/**
 * The unique ID of the integration.
 * @member {String} integrationId
 */
DefaultResponderDefaultResponder.prototype['integrationId'] = undefined;

/**
 * The type of the integration.
 * @member {String} integrationType
 */
DefaultResponderDefaultResponder.prototype['integrationType'] = undefined;

/**
 * Indicates whether the switchboard should deliver standby events.
 * @member {Boolean} deliverStandbyEvents
 */
DefaultResponderDefaultResponder.prototype['deliverStandbyEvents'] = undefined;

/**
 * The unique ID of the next switchboard integration.
 * @member {String} nextSwitchboardIntegrationId
 */
DefaultResponderDefaultResponder.prototype['nextSwitchboardIntegrationId'] = undefined;

/**
 * The number of messages to keep in the message history.
 * @member {Number} messageHistoryCount
 */
DefaultResponderDefaultResponder.prototype['messageHistoryCount'] = undefined;

/**
 * Indicates whether the default responder is inherited from the switchboard's global config or not. Returns `false` if a per-channel responder override has been set for this integration, and `true` otherwise.
 * @member {Boolean} inherited
 */
DefaultResponderDefaultResponder.prototype['inherited'] = undefined;
var _default = exports["default"] = DefaultResponderDefaultResponder;
