// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/DownloadMessageRefBodyAllOfApple.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _DownloadMessageRefBodyAllOfAppleInteractiveDataRef = _interopRequireDefault(require("./DownloadMessageRefBodyAllOfAppleInteractiveDataRef"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(a, n) { if (!(a instanceof n)) throw new TypeError("Cannot call a class as a function"); }
function _defineProperties(e, r) { for (var t = 0; t < r.length; t++) { var o = r[t]; o.enumerable = o.enumerable || !1, o.configurable = !0, "value" in o && (o.writable = !0), Object.defineProperty(e, _toPropertyKey(o.key), o); } }
function _createClass(e, r, t) { return r && _defineProperties(e.prototype, r), t && _defineProperties(e, t), Object.defineProperty(e, "prototype", { writable: !1 }), e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Sunshine Conversations API
 *
 * The version of the OpenAPI document: 16.0.1
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 *
 * OpenAPI Generator version: 4.3.1
 *
 * Do not edit the class manually.
 *
 */
/**
 * The DownloadMessageRefBodyAllOfApple model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/DownloadMessageRefBodyAllOfApple
 * @version 16.0.1
 */
var DownloadMessageRefBodyAllOfApple = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>DownloadMessageRefBodyAllOfApple</code>.
   * The message ref returned by Apple.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/DownloadMessageRefBodyAllOfApple
   * @param interactiveDataRef {module:sunshine-conversations-client/sunshine-conversations-client.model/DownloadMessageRefBodyAllOfAppleInteractiveDataRef} 
   */
  function DownloadMessageRefBodyAllOfApple(interactiveDataRef) {
    _classCallCheck(this, DownloadMessageRefBodyAllOfApple);
    DownloadMessageRefBodyAllOfApple.initialize(this, interactiveDataRef);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  return _createClass(DownloadMessageRefBodyAllOfApple, [{
    key: "getInteractiveDataRef",
    value:
    /**
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/DownloadMessageRefBodyAllOfAppleInteractiveDataRef}
         */
    function getInteractiveDataRef() {
      return this.interactiveDataRef;
    }

    /**
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/DownloadMessageRefBodyAllOfAppleInteractiveDataRef} interactiveDataRef
     */
  }, {
    key: "setInteractiveDataRef",
    value: function setInteractiveDataRef(interactiveDataRef) {
      this['interactiveDataRef'] = interactiveDataRef;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj, interactiveDataRef) {
      obj['interactiveDataRef'] = interactiveDataRef;
    }

    /**
     * Constructs a <code>DownloadMessageRefBodyAllOfApple</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/DownloadMessageRefBodyAllOfApple} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/DownloadMessageRefBodyAllOfApple} The populated <code>DownloadMessageRefBodyAllOfApple</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new DownloadMessageRefBodyAllOfApple();
        if (data.hasOwnProperty('interactiveDataRef')) {
          obj['interactiveDataRef'] = _DownloadMessageRefBodyAllOfAppleInteractiveDataRef["default"].constructFromObject(data['interactiveDataRef']);
        }
      }
      return obj;
    }
  }]);
}();
/**
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/DownloadMessageRefBodyAllOfAppleInteractiveDataRef} interactiveDataRef
 */
DownloadMessageRefBodyAllOfApple.prototype['interactiveDataRef'] = undefined;
var _default = exports["default"] = DownloadMessageRefBodyAllOfApple;
