// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/DownloadMessageRefBodyAllOfAppleInteractiveDataRef.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(a, n) { if (!(a instanceof n)) throw new TypeError("Cannot call a class as a function"); }
function _defineProperties(e, r) { for (var t = 0; t < r.length; t++) { var o = r[t]; o.enumerable = o.enumerable || !1, o.configurable = !0, "value" in o && (o.writable = !0), Object.defineProperty(e, _toPropertyKey(o.key), o); } }
function _createClass(e, r, t) { return r && _defineProperties(e.prototype, r), t && _defineProperties(e, t), Object.defineProperty(e, "prototype", { writable: !1 }), e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Sunshine Conversations API
 *
 * The version of the OpenAPI document: 16.0.1
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 *
 * OpenAPI Generator version: 4.3.1
 *
 * Do not edit the class manually.
 *
 */
/**
 * The DownloadMessageRefBodyAllOfAppleInteractiveDataRef model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/DownloadMessageRefBodyAllOfAppleInteractiveDataRef
 * @version 16.0.1
 */
var DownloadMessageRefBodyAllOfAppleInteractiveDataRef = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>DownloadMessageRefBodyAllOfAppleInteractiveDataRef</code>.
   * interactiveDataRef can be found in the passthrough webhook.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/DownloadMessageRefBodyAllOfAppleInteractiveDataRef
   * @param url {String} 
   * @param bid {String} 
   * @param key {String} 
   * @param signature {String} 
   * @param owner {String} 
   */
  function DownloadMessageRefBodyAllOfAppleInteractiveDataRef(url, bid, key, signature, owner) {
    _classCallCheck(this, DownloadMessageRefBodyAllOfAppleInteractiveDataRef);
    DownloadMessageRefBodyAllOfAppleInteractiveDataRef.initialize(this, url, bid, key, signature, owner);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  return _createClass(DownloadMessageRefBodyAllOfAppleInteractiveDataRef, [{
    key: "getUrl",
    value:
    /**
         * @return {String}
         */
    function getUrl() {
      return this.url;
    }

    /**
     * @param {String} url
     */
  }, {
    key: "setUrl",
    value: function setUrl(url) {
      this['url'] = url;
    }
    /**
         * @return {String}
         */
  }, {
    key: "getBid",
    value: function getBid() {
      return this.bid;
    }

    /**
     * @param {String} bid
     */
  }, {
    key: "setBid",
    value: function setBid(bid) {
      this['bid'] = bid;
    }
    /**
         * @return {String}
         */
  }, {
    key: "getKey",
    value: function getKey() {
      return this.key;
    }

    /**
     * @param {String} key
     */
  }, {
    key: "setKey",
    value: function setKey(key) {
      this['key'] = key;
    }
    /**
         * @return {String}
         */
  }, {
    key: "getSignature",
    value: function getSignature() {
      return this.signature;
    }

    /**
     * @param {String} signature
     */
  }, {
    key: "setSignature",
    value: function setSignature(signature) {
      this['signature'] = signature;
    }
    /**
         * @return {String}
         */
  }, {
    key: "getOwner",
    value: function getOwner() {
      return this.owner;
    }

    /**
     * @param {String} owner
     */
  }, {
    key: "setOwner",
    value: function setOwner(owner) {
      this['owner'] = owner;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj, url, bid, key, signature, owner) {
      obj['url'] = url;
      obj['bid'] = bid;
      obj['key'] = key;
      obj['signature'] = signature;
      obj['owner'] = owner;
    }

    /**
     * Constructs a <code>DownloadMessageRefBodyAllOfAppleInteractiveDataRef</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/DownloadMessageRefBodyAllOfAppleInteractiveDataRef} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/DownloadMessageRefBodyAllOfAppleInteractiveDataRef} The populated <code>DownloadMessageRefBodyAllOfAppleInteractiveDataRef</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new DownloadMessageRefBodyAllOfAppleInteractiveDataRef();
        if (data.hasOwnProperty('url')) {
          obj['url'] = _ApiClient["default"].convertToType(data['url'], 'String');
        }
        if (data.hasOwnProperty('bid')) {
          obj['bid'] = _ApiClient["default"].convertToType(data['bid'], 'String');
        }
        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
        }
        if (data.hasOwnProperty('signature')) {
          obj['signature'] = _ApiClient["default"].convertToType(data['signature'], 'String');
        }
        if (data.hasOwnProperty('owner')) {
          obj['owner'] = _ApiClient["default"].convertToType(data['owner'], 'String');
        }
      }
      return obj;
    }
  }]);
}();
/**
 * @member {String} url
 */
DownloadMessageRefBodyAllOfAppleInteractiveDataRef.prototype['url'] = undefined;

/**
 * @member {String} bid
 */
DownloadMessageRefBodyAllOfAppleInteractiveDataRef.prototype['bid'] = undefined;

/**
 * @member {String} key
 */
DownloadMessageRefBodyAllOfAppleInteractiveDataRef.prototype['key'] = undefined;

/**
 * @member {String} signature
 */
DownloadMessageRefBodyAllOfAppleInteractiveDataRef.prototype['signature'] = undefined;

/**
 * @member {String} owner
 */
DownloadMessageRefBodyAllOfAppleInteractiveDataRef.prototype['owner'] = undefined;
var _default = exports["default"] = DownloadMessageRefBodyAllOfAppleInteractiveDataRef;
