// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/DownloadMessageRefBody.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _DownloadMessageRefBodyAllOf = _interopRequireDefault(require("./DownloadMessageRefBodyAllOf"));
var _DownloadMessageRefBodyAllOfApple = _interopRequireDefault(require("./DownloadMessageRefBodyAllOfApple"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(a, n) { if (!(a instanceof n)) throw new TypeError("Cannot call a class as a function"); }
function _defineProperties(e, r) { for (var t = 0; t < r.length; t++) { var o = r[t]; o.enumerable = o.enumerable || !1, o.configurable = !0, "value" in o && (o.writable = !0), Object.defineProperty(e, _toPropertyKey(o.key), o); } }
function _createClass(e, r, t) { return r && _defineProperties(e.prototype, r), t && _defineProperties(e, t), Object.defineProperty(e, "prototype", { writable: !1 }), e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Sunshine Conversations API
 *
 * The version of the OpenAPI document: 16.0.1
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 *
 * OpenAPI Generator version: 4.3.1
 *
 * Do not edit the class manually.
 *
 */
/**
 * The DownloadMessageRefBody model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/DownloadMessageRefBody
 * @version 16.0.1
 */
var DownloadMessageRefBody = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>DownloadMessageRefBody</code>.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/DownloadMessageRefBody
   * @extends module:sunshine-conversations-client/sunshine-conversations-client.model/DownloadMessageRefBodyAllOf
   * @implements module:sunshine-conversations-client/sunshine-conversations-client.model/DownloadMessageRefBodyAllOf
   * @param userId {String} The id of the user.
   */
  function DownloadMessageRefBody(userId) {
    _classCallCheck(this, DownloadMessageRefBody);
    _DownloadMessageRefBodyAllOf["default"].initialize(this);
    DownloadMessageRefBody.initialize(this, userId);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  return _createClass(DownloadMessageRefBody, null, [{
    key: "initialize",
    value: function initialize(obj, userId) {}

    /**
     * Constructs a <code>DownloadMessageRefBody</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/DownloadMessageRefBody} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/DownloadMessageRefBody} The populated <code>DownloadMessageRefBody</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new DownloadMessageRefBody();
        _DownloadMessageRefBodyAllOf["default"].constructFromObject(data, obj);
        _DownloadMessageRefBodyAllOf["default"].constructFromObject(data, obj);
      }
      return obj;
    }
  }]);
}(); // Implement DownloadMessageRefBodyAllOf interface:
/**
 * The id of the user.
 * @member {String} userId
 */
_DownloadMessageRefBodyAllOf["default"].prototype['userId'] = undefined;
/**
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/DownloadMessageRefBodyAllOfApple} apple
 */
_DownloadMessageRefBodyAllOf["default"].prototype['apple'] = undefined;
var _default = exports["default"] = DownloadMessageRefBody;
