// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/FileMessage.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(a, n) { if (!(a instanceof n)) throw new TypeError("Cannot call a class as a function"); }
function _defineProperties(e, r) { for (var t = 0; t < r.length; t++) { var o = r[t]; o.enumerable = o.enumerable || !1, o.configurable = !0, "value" in o && (o.writable = !0), Object.defineProperty(e, _toPropertyKey(o.key), o); } }
function _createClass(e, r, t) { return r && _defineProperties(e.prototype, r), t && _defineProperties(e, t), Object.defineProperty(e, "prototype", { writable: !1 }), e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Sunshine Conversations API
 *
 * The version of the OpenAPI document: 16.0.1
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 *
 * OpenAPI Generator version: 4.3.1
 *
 * Do not edit the class manually.
 *
 */
/**
 * The FileMessage model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/FileMessage
 * @version 16.0.1
 */
var FileMessage = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>FileMessage</code>.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/FileMessage
   * @param type {String} The type of message.
   * @param mediaUrl {String} The URL for media, such as an image, attached to the message. <aside class=\"notice\"><strong>Note:</strong> An authorization header is required to access the mediaUrl when private attachments are enabled. See [configuring private attachments for messaging](https://developer.zendesk.com/documentation/zendesk-web-widget-sdks/messaging_private_attachments/) guide for more details.</aside> 
   */
  function FileMessage(type, mediaUrl) {
    _classCallCheck(this, FileMessage);
    FileMessage.initialize(this, type, mediaUrl);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  return _createClass(FileMessage, [{
    key: "getType",
    value:
    /**
         * Returns The type of message.
         * @return {String}
         */
    function getType() {
      return this.type;
    }

    /**
     * Sets The type of message.
     * @param {String} type The type of message.
     */
  }, {
    key: "setType",
    value: function setType(type) {
      this['type'] = type;
    }
    /**
         * Returns The URL for media, such as an image, attached to the message. <aside class=\"notice\"><strong>Note:</strong> An authorization header is required to access the mediaUrl when private attachments are enabled. See [configuring private attachments for messaging](https://developer.zendesk.com/documentation/zendesk-web-widget-sdks/messaging_private_attachments/) guide for more details.</aside> 
         * @return {String}
         */
  }, {
    key: "getMediaUrl",
    value: function getMediaUrl() {
      return this.mediaUrl;
    }

    /**
     * Sets The URL for media, such as an image, attached to the message. <aside class=\"notice\"><strong>Note:</strong> An authorization header is required to access the mediaUrl when private attachments are enabled. See [configuring private attachments for messaging](https://developer.zendesk.com/documentation/zendesk-web-widget-sdks/messaging_private_attachments/) guide for more details.</aside> 
     * @param {String} mediaUrl The URL for media, such as an image, attached to the message. <aside class=\"notice\"><strong>Note:</strong> An authorization header is required to access the mediaUrl when private attachments are enabled. See [configuring private attachments for messaging](https://developer.zendesk.com/documentation/zendesk-web-widget-sdks/messaging_private_attachments/) guide for more details.</aside> 
     */
  }, {
    key: "setMediaUrl",
    value: function setMediaUrl(mediaUrl) {
      this['mediaUrl'] = mediaUrl;
    }
    /**
         * Returns The size of the media.
         * @return {Number}
         */
  }, {
    key: "getMediaSize",
    value: function getMediaSize() {
      return this.mediaSize;
    }

    /**
     * Sets The size of the media.
     * @param {Number} mediaSize The size of the media.
     */
  }, {
    key: "setMediaSize",
    value: function setMediaSize(mediaSize) {
      this['mediaSize'] = mediaSize;
    }
    /**
         * Returns The media type of the file.
         * @return {String}
         */
  }, {
    key: "getMediaType",
    value: function getMediaType() {
      return this.mediaType;
    }

    /**
     * Sets The media type of the file.
     * @param {String} mediaType The media type of the file.
     */
  }, {
    key: "setMediaType",
    value: function setMediaType(mediaType) {
      this['mediaType'] = mediaType;
    }
    /**
         * Returns An optional description of the file for accessibility purposes. The field will be saved by default with the file name as the value.
         * @return {String}
         */
  }, {
    key: "getAltText",
    value: function getAltText() {
      return this.altText;
    }

    /**
     * Sets An optional description of the file for accessibility purposes. The field will be saved by default with the file name as the value.
     * @param {String} altText An optional description of the file for accessibility purposes. The field will be saved by default with the file name as the value.
     */
  }, {
    key: "setAltText",
    value: function setAltText(altText) {
      this['altText'] = altText;
    }
    /**
         * Returns The text content of the message.
         * @return {String}
         */
  }, {
    key: "getText",
    value: function getText() {
      return this.text;
    }

    /**
     * Sets The text content of the message.
     * @param {String} text The text content of the message.
     */
  }, {
    key: "setText",
    value: function setText(text) {
      this['text'] = text;
    }
    /**
         * Returns When set to true, the chat input will be disabled on supported client implementations when the message is the most recent one in the history. Can be used for guided flows or to temporarily disable the user's ability to send messages in the conversation.
         * @return {Boolean}
         */
  }, {
    key: "getBlockChatInput",
    value: function getBlockChatInput() {
      return this.blockChatInput;
    }

    /**
     * Sets When set to true, the chat input will be disabled on supported client implementations when the message is the most recent one in the history. Can be used for guided flows or to temporarily disable the user's ability to send messages in the conversation.
     * @param {Boolean} blockChatInput When set to true, the chat input will be disabled on supported client implementations when the message is the most recent one in the history. Can be used for guided flows or to temporarily disable the user's ability to send messages in the conversation.
     */
  }, {
    key: "setBlockChatInput",
    value: function setBlockChatInput(blockChatInput) {
      this['blockChatInput'] = blockChatInput;
    }
    /**
         * Returns HTML text content of the message. Can be provided in place of `text`. Cannot be used with `markdownText`. If no `text` is provided, will be converted to `text` upon reception to be displayed on channels that do not support rich text. See [rich text](https://developer.zendesk.com/documentation/conversations/messaging-platform/programmable-conversations/structured-messages/#rich-text) documentation for more information.
         * @return {String}
         */
  }, {
    key: "getHtmlText",
    value: function getHtmlText() {
      return this.htmlText;
    }

    /**
     * Sets HTML text content of the message. Can be provided in place of `text`. Cannot be used with `markdownText`. If no `text` is provided, will be converted to `text` upon reception to be displayed on channels that do not support rich text. See [rich text](https://developer.zendesk.com/documentation/conversations/messaging-platform/programmable-conversations/structured-messages/#rich-text) documentation for more information.
     * @param {String} htmlText HTML text content of the message. Can be provided in place of `text`. Cannot be used with `markdownText`. If no `text` is provided, will be converted to `text` upon reception to be displayed on channels that do not support rich text. See [rich text](https://developer.zendesk.com/documentation/conversations/messaging-platform/programmable-conversations/structured-messages/#rich-text) documentation for more information.
     */
  }, {
    key: "setHtmlText",
    value: function setHtmlText(htmlText) {
      this['htmlText'] = htmlText;
    }
    /**
         * Returns Markdown text content of the message. Can be provided in place of `text`. Cannot be used with `htmlText`. Will be converted to `htmlText` upon reception. If converted `htmlText` exceeds 4096 characters, the message will be rejected. If no `text` is provided, will be converted to `text` upon reception to be displayed on channels that do not support rich text. See [rich text](https://developer.zendesk.com/documentation/conversations/messaging-platform/programmable-conversations/structured-messages/#rich-text) documentation for more information.
         * @return {String}
         */
  }, {
    key: "getMarkdownText",
    value: function getMarkdownText() {
      return this.markdownText;
    }

    /**
     * Sets Markdown text content of the message. Can be provided in place of `text`. Cannot be used with `htmlText`. Will be converted to `htmlText` upon reception. If converted `htmlText` exceeds 4096 characters, the message will be rejected. If no `text` is provided, will be converted to `text` upon reception to be displayed on channels that do not support rich text. See [rich text](https://developer.zendesk.com/documentation/conversations/messaging-platform/programmable-conversations/structured-messages/#rich-text) documentation for more information.
     * @param {String} markdownText Markdown text content of the message. Can be provided in place of `text`. Cannot be used with `htmlText`. Will be converted to `htmlText` upon reception. If converted `htmlText` exceeds 4096 characters, the message will be rejected. If no `text` is provided, will be converted to `text` upon reception to be displayed on channels that do not support rich text. See [rich text](https://developer.zendesk.com/documentation/conversations/messaging-platform/programmable-conversations/structured-messages/#rich-text) documentation for more information.
     */
  }, {
    key: "setMarkdownText",
    value: function setMarkdownText(markdownText) {
      this['markdownText'] = markdownText;
    }
    /**
         * Returns An identifier used by Sunshine Conversations for internal purposes.
         * @return {String}
         */
  }, {
    key: "getAttachmentId",
    value: function getAttachmentId() {
      return this.attachmentId;
    }

    /**
     * Sets An identifier used by Sunshine Conversations for internal purposes.
     * @param {String} attachmentId An identifier used by Sunshine Conversations for internal purposes.
     */
  }, {
    key: "setAttachmentId",
    value: function setAttachmentId(attachmentId) {
      this['attachmentId'] = attachmentId;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj, type, mediaUrl) {
      obj['type'] = type;
      obj['mediaUrl'] = mediaUrl;
    }

    /**
     * Constructs a <code>FileMessage</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/FileMessage} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/FileMessage} The populated <code>FileMessage</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new FileMessage();
        if (data.hasOwnProperty('type')) {
          obj['type'] = _ApiClient["default"].convertToType(data['type'], 'String');
        }
        if (data.hasOwnProperty('mediaUrl')) {
          obj['mediaUrl'] = _ApiClient["default"].convertToType(data['mediaUrl'], 'String');
        }
        if (data.hasOwnProperty('mediaSize')) {
          obj['mediaSize'] = _ApiClient["default"].convertToType(data['mediaSize'], 'Number');
        }
        if (data.hasOwnProperty('mediaType')) {
          obj['mediaType'] = _ApiClient["default"].convertToType(data['mediaType'], 'String');
        }
        if (data.hasOwnProperty('altText')) {
          obj['altText'] = _ApiClient["default"].convertToType(data['altText'], 'String');
        }
        if (data.hasOwnProperty('text')) {
          obj['text'] = _ApiClient["default"].convertToType(data['text'], 'String');
        }
        if (data.hasOwnProperty('blockChatInput')) {
          obj['blockChatInput'] = _ApiClient["default"].convertToType(data['blockChatInput'], 'Boolean');
        }
        if (data.hasOwnProperty('htmlText')) {
          obj['htmlText'] = _ApiClient["default"].convertToType(data['htmlText'], 'String');
        }
        if (data.hasOwnProperty('markdownText')) {
          obj['markdownText'] = _ApiClient["default"].convertToType(data['markdownText'], 'String');
        }
        if (data.hasOwnProperty('attachmentId')) {
          obj['attachmentId'] = _ApiClient["default"].convertToType(data['attachmentId'], 'String');
        }
      }
      return obj;
    }
  }]);
}();
/**
 * The type of message.
 * @member {String} type
 * @default 'file'
 */
FileMessage.prototype['type'] = 'file';

/**
 * The URL for media, such as an image, attached to the message. <aside class=\"notice\"><strong>Note:</strong> An authorization header is required to access the mediaUrl when private attachments are enabled. See [configuring private attachments for messaging](https://developer.zendesk.com/documentation/zendesk-web-widget-sdks/messaging_private_attachments/) guide for more details.</aside> 
 * @member {String} mediaUrl
 */
FileMessage.prototype['mediaUrl'] = undefined;

/**
 * The size of the media.
 * @member {Number} mediaSize
 */
FileMessage.prototype['mediaSize'] = undefined;

/**
 * The media type of the file.
 * @member {String} mediaType
 */
FileMessage.prototype['mediaType'] = undefined;

/**
 * An optional description of the file for accessibility purposes. The field will be saved by default with the file name as the value.
 * @member {String} altText
 */
FileMessage.prototype['altText'] = undefined;

/**
 * The text content of the message.
 * @member {String} text
 */
FileMessage.prototype['text'] = undefined;

/**
 * When set to true, the chat input will be disabled on supported client implementations when the message is the most recent one in the history. Can be used for guided flows or to temporarily disable the user's ability to send messages in the conversation.
 * @member {Boolean} blockChatInput
 */
FileMessage.prototype['blockChatInput'] = undefined;

/**
 * HTML text content of the message. Can be provided in place of `text`. Cannot be used with `markdownText`. If no `text` is provided, will be converted to `text` upon reception to be displayed on channels that do not support rich text. See [rich text](https://developer.zendesk.com/documentation/conversations/messaging-platform/programmable-conversations/structured-messages/#rich-text) documentation for more information.
 * @member {String} htmlText
 */
FileMessage.prototype['htmlText'] = undefined;

/**
 * Markdown text content of the message. Can be provided in place of `text`. Cannot be used with `htmlText`. Will be converted to `htmlText` upon reception. If converted `htmlText` exceeds 4096 characters, the message will be rejected. If no `text` is provided, will be converted to `text` upon reception to be displayed on channels that do not support rich text. See [rich text](https://developer.zendesk.com/documentation/conversations/messaging-platform/programmable-conversations/structured-messages/#rich-text) documentation for more information.
 * @member {String} markdownText
 */
FileMessage.prototype['markdownText'] = undefined;

/**
 * An identifier used by Sunshine Conversations for internal purposes.
 * @member {String} attachmentId
 */
FileMessage.prototype['attachmentId'] = undefined;
var _default = exports["default"] = FileMessage;
