// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/Instagram.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _DefaultResponder = _interopRequireDefault(require("./DefaultResponder"));
var _DefaultResponderDefaultResponder = _interopRequireDefault(require("./DefaultResponderDefaultResponder"));
var _DefaultResponderId = _interopRequireDefault(require("./DefaultResponderId"));
var _InstagramAllOf = _interopRequireDefault(require("./InstagramAllOf"));
var _Integration = _interopRequireDefault(require("./Integration"));
var _Status = _interopRequireDefault(require("./Status"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(a, n) { if (!(a instanceof n)) throw new TypeError("Cannot call a class as a function"); }
function _defineProperties(e, r) { for (var t = 0; t < r.length; t++) { var o = r[t]; o.enumerable = o.enumerable || !1, o.configurable = !0, "value" in o && (o.writable = !0), Object.defineProperty(e, _toPropertyKey(o.key), o); } }
function _createClass(e, r, t) { return r && _defineProperties(e.prototype, r), t && _defineProperties(e, t), Object.defineProperty(e, "prototype", { writable: !1 }), e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Sunshine Conversations API
 *
 * The version of the OpenAPI document: 16.0.1
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 *
 * OpenAPI Generator version: 4.3.1
 *
 * Do not edit the class manually.
 *
 */
/**
 * The Instagram model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/Instagram
 * @version 16.0.1
 */
var Instagram = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Instagram</code>.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/Instagram
   * @extends module:sunshine-conversations-client/sunshine-conversations-client.model/Integration
   * @implements module:sunshine-conversations-client/sunshine-conversations-client.model/Integration
   * @implements module:sunshine-conversations-client/sunshine-conversations-client.model/InstagramAllOf
   * @implements module:sunshine-conversations-client/sunshine-conversations-client.model/DefaultResponderId
   * @implements module:sunshine-conversations-client/sunshine-conversations-client.model/DefaultResponder
   * @param type {String} Instagram Direct setup steps:   - Take note of your Facebook app ID and secret (apps can be created at [developer.facebook.com](https://developer.facebook.com));   - The Facebook app must have been submitted to Meta for app review with the \"pages_manage_metadata\" (to retrieve Page Access Tokens for the Pages and apps that the app user administers and to set a webhook), \"instagram_basic\", and \"instagram_manage_messages\" (to retrieve basic Instagram account information and send messages) permissions.   - In order to integrate an Instagram Direct app, you must acquire a Page Access Token from your user. Once you have acquired a page access token from your user, call the Create Integration endpoint with your app secret and ID and the user’s page access token. 
   * @param pageAccessToken {String} The Facebook Page Access Token of the Facebook page that is linked to your Instagram account.
   * @param appId {String} Your Facebook App ID.
   * @param appSecret {String} Your Facebook App secret.
   */
  function Instagram(type, pageAccessToken, appId, appSecret) {
    _classCallCheck(this, Instagram);
    _Integration["default"].initialize(this, type);
    _InstagramAllOf["default"].initialize(this, pageAccessToken, appId, appSecret);
    _DefaultResponderId["default"].initialize(this);
    _DefaultResponder["default"].initialize(this);
    Instagram.initialize(this, type, pageAccessToken, appId, appSecret);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  return _createClass(Instagram, [{
    key: "getType",
    value:
    /**
         * Returns Instagram Direct setup steps:   - Take note of your Facebook app ID and secret (apps can be created at [developer.facebook.com](https://developer.facebook.com));   - The Facebook app must have been submitted to Meta for app review with the \"pages_manage_metadata\" (to retrieve Page Access Tokens for the Pages and apps that the app user administers and to set a webhook), \"instagram_basic\", and \"instagram_manage_messages\" (to retrieve basic Instagram account information and send messages) permissions.   - In order to integrate an Instagram Direct app, you must acquire a Page Access Token from your user. Once you have acquired a page access token from your user, call the Create Integration endpoint with your app secret and ID and the user’s page access token. 
         * @return {String}
         */
    function getType() {
      return this.type;
    }

    /**
     * Sets Instagram Direct setup steps:   - Take note of your Facebook app ID and secret (apps can be created at [developer.facebook.com](https://developer.facebook.com));   - The Facebook app must have been submitted to Meta for app review with the \"pages_manage_metadata\" (to retrieve Page Access Tokens for the Pages and apps that the app user administers and to set a webhook), \"instagram_basic\", and \"instagram_manage_messages\" (to retrieve basic Instagram account information and send messages) permissions.   - In order to integrate an Instagram Direct app, you must acquire a Page Access Token from your user. Once you have acquired a page access token from your user, call the Create Integration endpoint with your app secret and ID and the user’s page access token. 
     * @param {String} type Instagram Direct setup steps:   - Take note of your Facebook app ID and secret (apps can be created at [developer.facebook.com](https://developer.facebook.com));   - The Facebook app must have been submitted to Meta for app review with the \"pages_manage_metadata\" (to retrieve Page Access Tokens for the Pages and apps that the app user administers and to set a webhook), \"instagram_basic\", and \"instagram_manage_messages\" (to retrieve basic Instagram account information and send messages) permissions.   - In order to integrate an Instagram Direct app, you must acquire a Page Access Token from your user. Once you have acquired a page access token from your user, call the Create Integration endpoint with your app secret and ID and the user’s page access token. 
     */
  }, {
    key: "setType",
    value: function setType(type) {
      this['type'] = type;
    }
    /**
         * Returns The Facebook Page Access Token of the Facebook page that is linked to your Instagram account.
         * @return {String}
         */
  }, {
    key: "getPageAccessToken",
    value: function getPageAccessToken() {
      return this.pageAccessToken;
    }

    /**
     * Sets The Facebook Page Access Token of the Facebook page that is linked to your Instagram account.
     * @param {String} pageAccessToken The Facebook Page Access Token of the Facebook page that is linked to your Instagram account.
     */
  }, {
    key: "setPageAccessToken",
    value: function setPageAccessToken(pageAccessToken) {
      this['pageAccessToken'] = pageAccessToken;
    }
    /**
         * Returns Your Facebook App ID.
         * @return {String}
         */
  }, {
    key: "getAppId",
    value: function getAppId() {
      return this.appId;
    }

    /**
     * Sets Your Facebook App ID.
     * @param {String} appId Your Facebook App ID.
     */
  }, {
    key: "setAppId",
    value: function setAppId(appId) {
      this['appId'] = appId;
    }
    /**
         * Returns Your Facebook App secret.
         * @return {String}
         */
  }, {
    key: "getAppSecret",
    value: function getAppSecret() {
      return this.appSecret;
    }

    /**
     * Sets Your Facebook App secret.
     * @param {String} appSecret Your Facebook App secret.
     */
  }, {
    key: "setAppSecret",
    value: function setAppSecret(appSecret) {
      this['appSecret'] = appSecret;
    }
    /**
         * Returns Your Instagram Business account name
         * @return {String}
         */
  }, {
    key: "getBusinessName",
    value: function getBusinessName() {
      return this.businessName;
    }

    /**
     * Sets Your Instagram Business account name
     * @param {String} businessName Your Instagram Business account name
     */
  }, {
    key: "setBusinessName",
    value: function setBusinessName(businessName) {
      this['businessName'] = businessName;
    }
    /**
         * Returns Your Instagram Business unique username
         * @return {String}
         */
  }, {
    key: "getBusinessUsername",
    value: function getBusinessUsername() {
      return this.businessUsername;
    }

    /**
     * Sets Your Instagram Business unique username
     * @param {String} businessUsername Your Instagram Business unique username
     */
  }, {
    key: "setBusinessUsername",
    value: function setBusinessUsername(businessUsername) {
      this['businessUsername'] = businessUsername;
    }
    /**
         * Returns The ID of the Facebook Page linked to your Instagram Business account
         * @return {String}
         */
  }, {
    key: "getPageId",
    value: function getPageId() {
      return this.pageId;
    }

    /**
     * Sets The ID of the Facebook Page linked to your Instagram Business account
     * @param {String} pageId The ID of the Facebook Page linked to your Instagram Business account
     */
  }, {
    key: "setPageId",
    value: function setPageId(pageId) {
      this['pageId'] = pageId;
    }
    /**
         * Returns The ID of the Instagram Business account
         * @return {String}
         */
  }, {
    key: "getBusinessId",
    value: function getBusinessId() {
      return this.businessId;
    }

    /**
     * Sets The ID of the Instagram Business account
     * @param {String} businessId The ID of the Instagram Business account
     */
  }, {
    key: "setBusinessId",
    value: function setBusinessId(businessId) {
      this['businessId'] = businessId;
    }
    /**
         * Returns The Facebook user's username. This is returned when integrating through the Dashboard
         * @return {String}
         */
  }, {
    key: "getUsername",
    value: function getUsername() {
      return this.username;
    }

    /**
     * Sets The Facebook user's username. This is returned when integrating through the Dashboard
     * @param {String} username The Facebook user's username. This is returned when integrating through the Dashboard
     */
  }, {
    key: "setUsername",
    value: function setUsername(username) {
      this['username'] = username;
    }
    /**
         * Returns The Facebook user's user ID. This is returned when integrating through the Dashboard
         * @return {String}
         */
  }, {
    key: "getUserId",
    value: function getUserId() {
      return this.userId;
    }

    /**
     * Sets The Facebook user's user ID. This is returned when integrating through the Dashboard
     * @param {String} userId The Facebook user's user ID. This is returned when integrating through the Dashboard
     */
  }, {
    key: "setUserId",
    value: function setUserId(userId) {
      this['userId'] = userId;
    }
    /**
         * Returns The default responder ID for the integration. This is the ID of the responder that will be used to send messages to the user. For more information, refer to the <a href=\"https://developer.zendesk.com/documentation/conversations/messaging-platform/programmable-conversations/switchboard/#default-integration-assignment\">Switchboard guide</a>. 
         * @return {String}
         */
  }, {
    key: "getDefaultResponderId",
    value: function getDefaultResponderId() {
      return this.defaultResponderId;
    }

    /**
     * Sets The default responder ID for the integration. This is the ID of the responder that will be used to send messages to the user. For more information, refer to the <a href=\"https://developer.zendesk.com/documentation/conversations/messaging-platform/programmable-conversations/switchboard/#default-integration-assignment\">Switchboard guide</a>. 
     * @param {String} defaultResponderId The default responder ID for the integration. This is the ID of the responder that will be used to send messages to the user. For more information, refer to the <a href=\"https://developer.zendesk.com/documentation/conversations/messaging-platform/programmable-conversations/switchboard/#default-integration-assignment\">Switchboard guide</a>. 
     */
  }, {
    key: "setDefaultResponderId",
    value: function setDefaultResponderId(defaultResponderId) {
      this['defaultResponderId'] = defaultResponderId;
    }
    /**
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/DefaultResponderDefaultResponder}
         */
  }, {
    key: "getDefaultResponder",
    value: function getDefaultResponder() {
      return this.defaultResponder;
    }

    /**
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/DefaultResponderDefaultResponder} defaultResponder
     */
  }, {
    key: "setDefaultResponder",
    value: function setDefaultResponder(defaultResponder) {
      this['defaultResponder'] = defaultResponder;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj, type, pageAccessToken, appId, appSecret) {
      obj['pageAccessToken'] = pageAccessToken;
      obj['appId'] = appId;
      obj['appSecret'] = appSecret;
    }

    /**
     * Constructs a <code>Instagram</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/Instagram} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/Instagram} The populated <code>Instagram</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Instagram();
        _Integration["default"].constructFromObject(data, obj);
        _Integration["default"].constructFromObject(data, obj);
        _InstagramAllOf["default"].constructFromObject(data, obj);
        _DefaultResponderId["default"].constructFromObject(data, obj);
        _DefaultResponder["default"].constructFromObject(data, obj);
        if (data.hasOwnProperty('type')) {
          obj['type'] = _ApiClient["default"].convertToType(data['type'], 'String');
        }
        if (data.hasOwnProperty('pageAccessToken')) {
          obj['pageAccessToken'] = _ApiClient["default"].convertToType(data['pageAccessToken'], 'String');
        }
        if (data.hasOwnProperty('appId')) {
          obj['appId'] = _ApiClient["default"].convertToType(data['appId'], 'String');
        }
        if (data.hasOwnProperty('appSecret')) {
          obj['appSecret'] = _ApiClient["default"].convertToType(data['appSecret'], 'String');
        }
        if (data.hasOwnProperty('businessName')) {
          obj['businessName'] = _ApiClient["default"].convertToType(data['businessName'], 'String');
        }
        if (data.hasOwnProperty('businessUsername')) {
          obj['businessUsername'] = _ApiClient["default"].convertToType(data['businessUsername'], 'String');
        }
        if (data.hasOwnProperty('pageId')) {
          obj['pageId'] = _ApiClient["default"].convertToType(data['pageId'], 'String');
        }
        if (data.hasOwnProperty('businessId')) {
          obj['businessId'] = _ApiClient["default"].convertToType(data['businessId'], 'String');
        }
        if (data.hasOwnProperty('username')) {
          obj['username'] = _ApiClient["default"].convertToType(data['username'], 'String');
        }
        if (data.hasOwnProperty('userId')) {
          obj['userId'] = _ApiClient["default"].convertToType(data['userId'], 'String');
        }
        if (data.hasOwnProperty('defaultResponderId')) {
          obj['defaultResponderId'] = _ApiClient["default"].convertToType(data['defaultResponderId'], 'String');
        }
        if (data.hasOwnProperty('defaultResponder')) {
          obj['defaultResponder'] = _DefaultResponderDefaultResponder["default"].constructFromObject(data['defaultResponder']);
        }
      }
      return obj;
    }
  }]);
}();
/**
 * Instagram Direct setup steps:   - Take note of your Facebook app ID and secret (apps can be created at [developer.facebook.com](https://developer.facebook.com));   - The Facebook app must have been submitted to Meta for app review with the \"pages_manage_metadata\" (to retrieve Page Access Tokens for the Pages and apps that the app user administers and to set a webhook), \"instagram_basic\", and \"instagram_manage_messages\" (to retrieve basic Instagram account information and send messages) permissions.   - In order to integrate an Instagram Direct app, you must acquire a Page Access Token from your user. Once you have acquired a page access token from your user, call the Create Integration endpoint with your app secret and ID and the user’s page access token. 
 * @member {String} type
 * @default 'instagram'
 */
Instagram.prototype['type'] = 'instagram';

/**
 * The Facebook Page Access Token of the Facebook page that is linked to your Instagram account.
 * @member {String} pageAccessToken
 */
Instagram.prototype['pageAccessToken'] = undefined;

/**
 * Your Facebook App ID.
 * @member {String} appId
 */
Instagram.prototype['appId'] = undefined;

/**
 * Your Facebook App secret.
 * @member {String} appSecret
 */
Instagram.prototype['appSecret'] = undefined;

/**
 * Your Instagram Business account name
 * @member {String} businessName
 */
Instagram.prototype['businessName'] = undefined;

/**
 * Your Instagram Business unique username
 * @member {String} businessUsername
 */
Instagram.prototype['businessUsername'] = undefined;

/**
 * The ID of the Facebook Page linked to your Instagram Business account
 * @member {String} pageId
 */
Instagram.prototype['pageId'] = undefined;

/**
 * The ID of the Instagram Business account
 * @member {String} businessId
 */
Instagram.prototype['businessId'] = undefined;

/**
 * The Facebook user's username. This is returned when integrating through the Dashboard
 * @member {String} username
 */
Instagram.prototype['username'] = undefined;

/**
 * The Facebook user's user ID. This is returned when integrating through the Dashboard
 * @member {String} userId
 */
Instagram.prototype['userId'] = undefined;

/**
 * The default responder ID for the integration. This is the ID of the responder that will be used to send messages to the user. For more information, refer to the <a href=\"https://developer.zendesk.com/documentation/conversations/messaging-platform/programmable-conversations/switchboard/#default-integration-assignment\">Switchboard guide</a>. 
 * @member {String} defaultResponderId
 */
Instagram.prototype['defaultResponderId'] = undefined;

/**
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/DefaultResponderDefaultResponder} defaultResponder
 */
Instagram.prototype['defaultResponder'] = undefined;

// Implement Integration interface:
/**
 * The unique ID of the integration.
 * @member {String} id
 */
_Integration["default"].prototype['id'] = undefined;
/**
 * @member {String} type
 */
_Integration["default"].prototype['type'] = undefined;
/**
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/Status} status
 */
_Integration["default"].prototype['status'] = undefined;
/**
 * A human-friendly name used to identify the integration.
 * @member {String} displayName
 */
_Integration["default"].prototype['displayName'] = undefined;
// Implement InstagramAllOf interface:
/**
 * Instagram Direct setup steps:   - Take note of your Facebook app ID and secret (apps can be created at [developer.facebook.com](https://developer.facebook.com));   - The Facebook app must have been submitted to Meta for app review with the \"pages_manage_metadata\" (to retrieve Page Access Tokens for the Pages and apps that the app user administers and to set a webhook), \"instagram_basic\", and \"instagram_manage_messages\" (to retrieve basic Instagram account information and send messages) permissions.   - In order to integrate an Instagram Direct app, you must acquire a Page Access Token from your user. Once you have acquired a page access token from your user, call the Create Integration endpoint with your app secret and ID and the user’s page access token. 
 * @member {String} type
 * @default 'instagram'
 */
_InstagramAllOf["default"].prototype['type'] = 'instagram';
/**
 * The Facebook Page Access Token of the Facebook page that is linked to your Instagram account.
 * @member {String} pageAccessToken
 */
_InstagramAllOf["default"].prototype['pageAccessToken'] = undefined;
/**
 * Your Facebook App ID.
 * @member {String} appId
 */
_InstagramAllOf["default"].prototype['appId'] = undefined;
/**
 * Your Facebook App secret.
 * @member {String} appSecret
 */
_InstagramAllOf["default"].prototype['appSecret'] = undefined;
/**
 * Your Instagram Business account name
 * @member {String} businessName
 */
_InstagramAllOf["default"].prototype['businessName'] = undefined;
/**
 * Your Instagram Business unique username
 * @member {String} businessUsername
 */
_InstagramAllOf["default"].prototype['businessUsername'] = undefined;
/**
 * The ID of the Facebook Page linked to your Instagram Business account
 * @member {String} pageId
 */
_InstagramAllOf["default"].prototype['pageId'] = undefined;
/**
 * The ID of the Instagram Business account
 * @member {String} businessId
 */
_InstagramAllOf["default"].prototype['businessId'] = undefined;
/**
 * The Facebook user's username. This is returned when integrating through the Dashboard
 * @member {String} username
 */
_InstagramAllOf["default"].prototype['username'] = undefined;
/**
 * The Facebook user's user ID. This is returned when integrating through the Dashboard
 * @member {String} userId
 */
_InstagramAllOf["default"].prototype['userId'] = undefined;
// Implement DefaultResponderId interface:
/**
 * The default responder ID for the integration. This is the ID of the responder that will be used to send messages to the user. For more information, refer to the <a href=\"https://developer.zendesk.com/documentation/conversations/messaging-platform/programmable-conversations/switchboard/#default-integration-assignment\">Switchboard guide</a>. 
 * @member {String} defaultResponderId
 */
_DefaultResponderId["default"].prototype['defaultResponderId'] = undefined;
// Implement DefaultResponder interface:
/**
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/DefaultResponderDefaultResponder} defaultResponder
 */
_DefaultResponder["default"].prototype['defaultResponder'] = undefined;
var _default = exports["default"] = Instagram;
