// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/ListMessage.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _ActionSubset = _interopRequireDefault(require("./ActionSubset"));
var _Item = _interopRequireDefault(require("./Item"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(a, n) { if (!(a instanceof n)) throw new TypeError("Cannot call a class as a function"); }
function _defineProperties(e, r) { for (var t = 0; t < r.length; t++) { var o = r[t]; o.enumerable = o.enumerable || !1, o.configurable = !0, "value" in o && (o.writable = !0), Object.defineProperty(e, _toPropertyKey(o.key), o); } }
function _createClass(e, r, t) { return r && _defineProperties(e.prototype, r), t && _defineProperties(e, t), Object.defineProperty(e, "prototype", { writable: !1 }), e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Sunshine Conversations API
 *
 * The version of the OpenAPI document: 16.0.1
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 *
 * OpenAPI Generator version: 4.3.1
 *
 * Do not edit the class manually.
 *
 */
/**
 * The ListMessage model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/ListMessage
 * @version 16.0.1
 */
var ListMessage = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ListMessage</code>.
   * List messages are a vertically scrollable set of items that may each contain text, an image, and message actions. Not all messaging channels fully support list messages. * Facebook Messenger and WeChat have native support. * For LINE and our Android, iOS and Web SDK, Sunshine Conversations converts list messages to carousel. * On WhatsApp, Telegram and Twitter, Sunshine Conversations converts list messages to multiple rich messages. * On all other platforms, Sunshine Conversations converts list messages to raw text. 
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/ListMessage
   * @param type {String} The type of message.
   * @param items {Array.<module:sunshine-conversations-client/sunshine-conversations-client.model/Item>} An array of objects representing the items associated with the message. Only present in carousel and list type messages.
   */
  function ListMessage(type, items) {
    _classCallCheck(this, ListMessage);
    ListMessage.initialize(this, type, items);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  return _createClass(ListMessage, [{
    key: "getType",
    value:
    /**
         * Returns The type of message.
         * @return {String}
         */
    function getType() {
      return this.type;
    }

    /**
     * Sets The type of message.
     * @param {String} type The type of message.
     */
  }, {
    key: "setType",
    value: function setType(type) {
      this['type'] = type;
    }
    /**
         * Returns When set to true, the chat input will be disabled on supported client implementations when the message is the most recent one in the history. Can be used for guided flows or to temporarily disable the user's ability to send messages in the conversation.
         * @return {Boolean}
         */
  }, {
    key: "getBlockChatInput",
    value: function getBlockChatInput() {
      return this.blockChatInput;
    }

    /**
     * Sets When set to true, the chat input will be disabled on supported client implementations when the message is the most recent one in the history. Can be used for guided flows or to temporarily disable the user's ability to send messages in the conversation.
     * @param {Boolean} blockChatInput When set to true, the chat input will be disabled on supported client implementations when the message is the most recent one in the history. Can be used for guided flows or to temporarily disable the user's ability to send messages in the conversation.
     */
  }, {
    key: "setBlockChatInput",
    value: function setBlockChatInput(blockChatInput) {
      this['blockChatInput'] = blockChatInput;
    }
    /**
         * Returns An array of objects representing the items associated with the message. Only present in carousel and list type messages.
         * @return {Array.<module:sunshine-conversations-client/sunshine-conversations-client.model/Item>}
         */
  }, {
    key: "getItems",
    value: function getItems() {
      return this.items;
    }

    /**
     * Sets An array of objects representing the items associated with the message. Only present in carousel and list type messages.
     * @param {Array.<module:sunshine-conversations-client/sunshine-conversations-client.model/Item>} items An array of objects representing the items associated with the message. Only present in carousel and list type messages.
     */
  }, {
    key: "setItems",
    value: function setItems(items) {
      this['items'] = items;
    }
    /**
         * Returns An array of objects representing the actions associated with the message. The array length is limited by the third party channel.
         * @return {Array.<module:sunshine-conversations-client/sunshine-conversations-client.model/ActionSubset>}
         */
  }, {
    key: "getActions",
    value: function getActions() {
      return this.actions;
    }

    /**
     * Sets An array of objects representing the actions associated with the message. The array length is limited by the third party channel.
     * @param {Array.<module:sunshine-conversations-client/sunshine-conversations-client.model/ActionSubset>} actions An array of objects representing the actions associated with the message. The array length is limited by the third party channel.
     */
  }, {
    key: "setActions",
    value: function setActions(actions) {
      this['actions'] = actions;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj, type, items) {
      obj['type'] = type;
      obj['items'] = items;
    }

    /**
     * Constructs a <code>ListMessage</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/ListMessage} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/ListMessage} The populated <code>ListMessage</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ListMessage();
        if (data.hasOwnProperty('type')) {
          obj['type'] = _ApiClient["default"].convertToType(data['type'], 'String');
        }
        if (data.hasOwnProperty('blockChatInput')) {
          obj['blockChatInput'] = _ApiClient["default"].convertToType(data['blockChatInput'], 'Boolean');
        }
        if (data.hasOwnProperty('items')) {
          obj['items'] = _ApiClient["default"].convertToType(data['items'], [_Item["default"]]);
        }
        if (data.hasOwnProperty('actions')) {
          obj['actions'] = _ApiClient["default"].convertToType(data['actions'], [_ActionSubset["default"]]);
        }
      }
      return obj;
    }
  }]);
}();
/**
 * The type of message.
 * @member {String} type
 * @default 'list'
 */
ListMessage.prototype['type'] = 'list';

/**
 * When set to true, the chat input will be disabled on supported client implementations when the message is the most recent one in the history. Can be used for guided flows or to temporarily disable the user's ability to send messages in the conversation.
 * @member {Boolean} blockChatInput
 */
ListMessage.prototype['blockChatInput'] = undefined;

/**
 * An array of objects representing the items associated with the message. Only present in carousel and list type messages.
 * @member {Array.<module:sunshine-conversations-client/sunshine-conversations-client.model/Item>} items
 */
ListMessage.prototype['items'] = undefined;

/**
 * An array of objects representing the actions associated with the message. The array length is limited by the third party channel.
 * @member {Array.<module:sunshine-conversations-client/sunshine-conversations-client.model/ActionSubset>} actions
 */
ListMessage.prototype['actions'] = undefined;
var _default = exports["default"] = ListMessage;
