// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/Messagebird.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _DefaultResponder = _interopRequireDefault(require("./DefaultResponder"));
var _DefaultResponderDefaultResponder = _interopRequireDefault(require("./DefaultResponderDefaultResponder"));
var _DefaultResponderId = _interopRequireDefault(require("./DefaultResponderId"));
var _Integration = _interopRequireDefault(require("./Integration"));
var _MessagebirdAllOf = _interopRequireDefault(require("./MessagebirdAllOf"));
var _Status = _interopRequireDefault(require("./Status"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(a, n) { if (!(a instanceof n)) throw new TypeError("Cannot call a class as a function"); }
function _defineProperties(e, r) { for (var t = 0; t < r.length; t++) { var o = r[t]; o.enumerable = o.enumerable || !1, o.configurable = !0, "value" in o && (o.writable = !0), Object.defineProperty(e, _toPropertyKey(o.key), o); } }
function _createClass(e, r, t) { return r && _defineProperties(e.prototype, r), t && _defineProperties(e, t), Object.defineProperty(e, "prototype", { writable: !1 }), e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Sunshine Conversations API
 *
 * The version of the OpenAPI document: 16.0.1
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 *
 * OpenAPI Generator version: 4.3.1
 *
 * Do not edit the class manually.
 *
 */
/**
 * The Messagebird model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/Messagebird
 * @version 16.0.1
 */
var Messagebird = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Messagebird</code>.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/Messagebird
   * @extends module:sunshine-conversations-client/sunshine-conversations-client.model/Integration
   * @implements module:sunshine-conversations-client/sunshine-conversations-client.model/Integration
   * @implements module:sunshine-conversations-client/sunshine-conversations-client.model/MessagebirdAllOf
   * @implements module:sunshine-conversations-client/sunshine-conversations-client.model/DefaultResponderId
   * @implements module:sunshine-conversations-client/sunshine-conversations-client.model/DefaultResponder
   * @param type {String} To configure a MessageBird integration, acquire the accessKey, the signingKey and the MessageBird number you would like to use, then call the Create Integration endpoint. The response will include the integration’s `_id` and `webhookSecret`, which must be used to configure the webhook in MessageBird. In the Flow Builder for the MessageBird number you’ve used to integrate, add a new step with the following settings: - Create a new Call HTTP endpoint with SMS flow. - Select your desired SMS number for Incoming SMS. - Click on Forward to URL and set its method to POST. - Then, using the integration _id and webhookSecret returned from the create integration call, enter the following into the URL field:  `https://app.smooch.io/api/messagebird/webhooks/{appId}/{integrationId}/{webhookSecret}` - Select application/json for the Set Content-Type header field. - Save and publish your changes. 
   * @param accessKey {String} The public API key of your MessageBird account.
   * @param signingKey {String} The signing key of your MessageBird account. Used to validate the webhooks' origin.
   * @param originator {String} Sunshine Conversations will receive all messages sent to this phone number.
   */
  function Messagebird(type, accessKey, signingKey, originator) {
    _classCallCheck(this, Messagebird);
    _Integration["default"].initialize(this, type);
    _MessagebirdAllOf["default"].initialize(this, accessKey, signingKey, originator);
    _DefaultResponderId["default"].initialize(this);
    _DefaultResponder["default"].initialize(this);
    Messagebird.initialize(this, type, accessKey, signingKey, originator);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  return _createClass(Messagebird, [{
    key: "getType",
    value:
    /**
         * Returns To configure a MessageBird integration, acquire the accessKey, the signingKey and the MessageBird number you would like to use, then call the Create Integration endpoint. The response will include the integration’s `_id` and `webhookSecret`, which must be used to configure the webhook in MessageBird. In the Flow Builder for the MessageBird number you’ve used to integrate, add a new step with the following settings: - Create a new Call HTTP endpoint with SMS flow. - Select your desired SMS number for Incoming SMS. - Click on Forward to URL and set its method to POST. - Then, using the integration _id and webhookSecret returned from the create integration call, enter the following into the URL field:  `https://app.smooch.io/api/messagebird/webhooks/{appId}/{integrationId}/{webhookSecret}` - Select application/json for the Set Content-Type header field. - Save and publish your changes. 
         * @return {String}
         */
    function getType() {
      return this.type;
    }

    /**
     * Sets To configure a MessageBird integration, acquire the accessKey, the signingKey and the MessageBird number you would like to use, then call the Create Integration endpoint. The response will include the integration’s `_id` and `webhookSecret`, which must be used to configure the webhook in MessageBird. In the Flow Builder for the MessageBird number you’ve used to integrate, add a new step with the following settings: - Create a new Call HTTP endpoint with SMS flow. - Select your desired SMS number for Incoming SMS. - Click on Forward to URL and set its method to POST. - Then, using the integration _id and webhookSecret returned from the create integration call, enter the following into the URL field:  `https://app.smooch.io/api/messagebird/webhooks/{appId}/{integrationId}/{webhookSecret}` - Select application/json for the Set Content-Type header field. - Save and publish your changes. 
     * @param {String} type To configure a MessageBird integration, acquire the accessKey, the signingKey and the MessageBird number you would like to use, then call the Create Integration endpoint. The response will include the integration’s `_id` and `webhookSecret`, which must be used to configure the webhook in MessageBird. In the Flow Builder for the MessageBird number you’ve used to integrate, add a new step with the following settings: - Create a new Call HTTP endpoint with SMS flow. - Select your desired SMS number for Incoming SMS. - Click on Forward to URL and set its method to POST. - Then, using the integration _id and webhookSecret returned from the create integration call, enter the following into the URL field:  `https://app.smooch.io/api/messagebird/webhooks/{appId}/{integrationId}/{webhookSecret}` - Select application/json for the Set Content-Type header field. - Save and publish your changes. 
     */
  }, {
    key: "setType",
    value: function setType(type) {
      this['type'] = type;
    }
    /**
         * Returns The public API key of your MessageBird account.
         * @return {String}
         */
  }, {
    key: "getAccessKey",
    value: function getAccessKey() {
      return this.accessKey;
    }

    /**
     * Sets The public API key of your MessageBird account.
     * @param {String} accessKey The public API key of your MessageBird account.
     */
  }, {
    key: "setAccessKey",
    value: function setAccessKey(accessKey) {
      this['accessKey'] = accessKey;
    }
    /**
         * Returns The signing key of your MessageBird account. Used to validate the webhooks' origin.
         * @return {String}
         */
  }, {
    key: "getSigningKey",
    value: function getSigningKey() {
      return this.signingKey;
    }

    /**
     * Sets The signing key of your MessageBird account. Used to validate the webhooks' origin.
     * @param {String} signingKey The signing key of your MessageBird account. Used to validate the webhooks' origin.
     */
  }, {
    key: "setSigningKey",
    value: function setSigningKey(signingKey) {
      this['signingKey'] = signingKey;
    }
    /**
         * Returns Sunshine Conversations will receive all messages sent to this phone number.
         * @return {String}
         */
  }, {
    key: "getOriginator",
    value: function getOriginator() {
      return this.originator;
    }

    /**
     * Sets Sunshine Conversations will receive all messages sent to this phone number.
     * @param {String} originator Sunshine Conversations will receive all messages sent to this phone number.
     */
  }, {
    key: "setOriginator",
    value: function setOriginator(originator) {
      this['originator'] = originator;
    }
    /**
         * Returns The secret that is used to configure webhooks in MessageBird.
         * @return {String}
         */
  }, {
    key: "getWebhookSecret",
    value: function getWebhookSecret() {
      return this.webhookSecret;
    }

    /**
     * Sets The secret that is used to configure webhooks in MessageBird.
     * @param {String} webhookSecret The secret that is used to configure webhooks in MessageBird.
     */
  }, {
    key: "setWebhookSecret",
    value: function setWebhookSecret(webhookSecret) {
      this['webhookSecret'] = webhookSecret;
    }
    /**
         * Returns The default responder ID for the integration. This is the ID of the responder that will be used to send messages to the user. For more information, refer to the <a href=\"https://developer.zendesk.com/documentation/conversations/messaging-platform/programmable-conversations/switchboard/#default-integration-assignment\">Switchboard guide</a>. 
         * @return {String}
         */
  }, {
    key: "getDefaultResponderId",
    value: function getDefaultResponderId() {
      return this.defaultResponderId;
    }

    /**
     * Sets The default responder ID for the integration. This is the ID of the responder that will be used to send messages to the user. For more information, refer to the <a href=\"https://developer.zendesk.com/documentation/conversations/messaging-platform/programmable-conversations/switchboard/#default-integration-assignment\">Switchboard guide</a>. 
     * @param {String} defaultResponderId The default responder ID for the integration. This is the ID of the responder that will be used to send messages to the user. For more information, refer to the <a href=\"https://developer.zendesk.com/documentation/conversations/messaging-platform/programmable-conversations/switchboard/#default-integration-assignment\">Switchboard guide</a>. 
     */
  }, {
    key: "setDefaultResponderId",
    value: function setDefaultResponderId(defaultResponderId) {
      this['defaultResponderId'] = defaultResponderId;
    }
    /**
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/DefaultResponderDefaultResponder}
         */
  }, {
    key: "getDefaultResponder",
    value: function getDefaultResponder() {
      return this.defaultResponder;
    }

    /**
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/DefaultResponderDefaultResponder} defaultResponder
     */
  }, {
    key: "setDefaultResponder",
    value: function setDefaultResponder(defaultResponder) {
      this['defaultResponder'] = defaultResponder;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj, type, accessKey, signingKey, originator) {
      obj['accessKey'] = accessKey;
      obj['signingKey'] = signingKey;
      obj['originator'] = originator;
    }

    /**
     * Constructs a <code>Messagebird</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/Messagebird} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/Messagebird} The populated <code>Messagebird</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Messagebird();
        _Integration["default"].constructFromObject(data, obj);
        _Integration["default"].constructFromObject(data, obj);
        _MessagebirdAllOf["default"].constructFromObject(data, obj);
        _DefaultResponderId["default"].constructFromObject(data, obj);
        _DefaultResponder["default"].constructFromObject(data, obj);
        if (data.hasOwnProperty('type')) {
          obj['type'] = _ApiClient["default"].convertToType(data['type'], 'String');
        }
        if (data.hasOwnProperty('accessKey')) {
          obj['accessKey'] = _ApiClient["default"].convertToType(data['accessKey'], 'String');
        }
        if (data.hasOwnProperty('signingKey')) {
          obj['signingKey'] = _ApiClient["default"].convertToType(data['signingKey'], 'String');
        }
        if (data.hasOwnProperty('originator')) {
          obj['originator'] = _ApiClient["default"].convertToType(data['originator'], 'String');
        }
        if (data.hasOwnProperty('webhookSecret')) {
          obj['webhookSecret'] = _ApiClient["default"].convertToType(data['webhookSecret'], 'String');
        }
        if (data.hasOwnProperty('defaultResponderId')) {
          obj['defaultResponderId'] = _ApiClient["default"].convertToType(data['defaultResponderId'], 'String');
        }
        if (data.hasOwnProperty('defaultResponder')) {
          obj['defaultResponder'] = _DefaultResponderDefaultResponder["default"].constructFromObject(data['defaultResponder']);
        }
      }
      return obj;
    }
  }]);
}();
/**
 * To configure a MessageBird integration, acquire the accessKey, the signingKey and the MessageBird number you would like to use, then call the Create Integration endpoint. The response will include the integration’s `_id` and `webhookSecret`, which must be used to configure the webhook in MessageBird. In the Flow Builder for the MessageBird number you’ve used to integrate, add a new step with the following settings: - Create a new Call HTTP endpoint with SMS flow. - Select your desired SMS number for Incoming SMS. - Click on Forward to URL and set its method to POST. - Then, using the integration _id and webhookSecret returned from the create integration call, enter the following into the URL field:  `https://app.smooch.io/api/messagebird/webhooks/{appId}/{integrationId}/{webhookSecret}` - Select application/json for the Set Content-Type header field. - Save and publish your changes. 
 * @member {String} type
 * @default 'messagebird'
 */
Messagebird.prototype['type'] = 'messagebird';

/**
 * The public API key of your MessageBird account.
 * @member {String} accessKey
 */
Messagebird.prototype['accessKey'] = undefined;

/**
 * The signing key of your MessageBird account. Used to validate the webhooks' origin.
 * @member {String} signingKey
 */
Messagebird.prototype['signingKey'] = undefined;

/**
 * Sunshine Conversations will receive all messages sent to this phone number.
 * @member {String} originator
 */
Messagebird.prototype['originator'] = undefined;

/**
 * The secret that is used to configure webhooks in MessageBird.
 * @member {String} webhookSecret
 */
Messagebird.prototype['webhookSecret'] = undefined;

/**
 * The default responder ID for the integration. This is the ID of the responder that will be used to send messages to the user. For more information, refer to the <a href=\"https://developer.zendesk.com/documentation/conversations/messaging-platform/programmable-conversations/switchboard/#default-integration-assignment\">Switchboard guide</a>. 
 * @member {String} defaultResponderId
 */
Messagebird.prototype['defaultResponderId'] = undefined;

/**
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/DefaultResponderDefaultResponder} defaultResponder
 */
Messagebird.prototype['defaultResponder'] = undefined;

// Implement Integration interface:
/**
 * The unique ID of the integration.
 * @member {String} id
 */
_Integration["default"].prototype['id'] = undefined;
/**
 * @member {String} type
 */
_Integration["default"].prototype['type'] = undefined;
/**
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/Status} status
 */
_Integration["default"].prototype['status'] = undefined;
/**
 * A human-friendly name used to identify the integration.
 * @member {String} displayName
 */
_Integration["default"].prototype['displayName'] = undefined;
// Implement MessagebirdAllOf interface:
/**
 * To configure a MessageBird integration, acquire the accessKey, the signingKey and the MessageBird number you would like to use, then call the Create Integration endpoint. The response will include the integration’s `_id` and `webhookSecret`, which must be used to configure the webhook in MessageBird. In the Flow Builder for the MessageBird number you’ve used to integrate, add a new step with the following settings: - Create a new Call HTTP endpoint with SMS flow. - Select your desired SMS number for Incoming SMS. - Click on Forward to URL and set its method to POST. - Then, using the integration _id and webhookSecret returned from the create integration call, enter the following into the URL field:  `https://app.smooch.io/api/messagebird/webhooks/{appId}/{integrationId}/{webhookSecret}` - Select application/json for the Set Content-Type header field. - Save and publish your changes. 
 * @member {String} type
 * @default 'messagebird'
 */
_MessagebirdAllOf["default"].prototype['type'] = 'messagebird';
/**
 * The public API key of your MessageBird account.
 * @member {String} accessKey
 */
_MessagebirdAllOf["default"].prototype['accessKey'] = undefined;
/**
 * The signing key of your MessageBird account. Used to validate the webhooks' origin.
 * @member {String} signingKey
 */
_MessagebirdAllOf["default"].prototype['signingKey'] = undefined;
/**
 * Sunshine Conversations will receive all messages sent to this phone number.
 * @member {String} originator
 */
_MessagebirdAllOf["default"].prototype['originator'] = undefined;
/**
 * The secret that is used to configure webhooks in MessageBird.
 * @member {String} webhookSecret
 */
_MessagebirdAllOf["default"].prototype['webhookSecret'] = undefined;
// Implement DefaultResponderId interface:
/**
 * The default responder ID for the integration. This is the ID of the responder that will be used to send messages to the user. For more information, refer to the <a href=\"https://developer.zendesk.com/documentation/conversations/messaging-platform/programmable-conversations/switchboard/#default-integration-assignment\">Switchboard guide</a>. 
 * @member {String} defaultResponderId
 */
_DefaultResponderId["default"].prototype['defaultResponderId'] = undefined;
// Implement DefaultResponder interface:
/**
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/DefaultResponderDefaultResponder} defaultResponder
 */
_DefaultResponder["default"].prototype['defaultResponder'] = undefined;
var _default = exports["default"] = Messagebird;
