// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/MessageOverride.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _AppleMessageOverridePayload = _interopRequireDefault(require("./AppleMessageOverridePayload"));
var _MessageOverrideApple = _interopRequireDefault(require("./MessageOverrideApple"));
var _MessageOverrideLine = _interopRequireDefault(require("./MessageOverrideLine"));
var _MessageOverrideMessenger = _interopRequireDefault(require("./MessageOverrideMessenger"));
var _MessageOverridePayload = _interopRequireDefault(require("./MessageOverridePayload"));
var _MessageOverrideWhatsapp = _interopRequireDefault(require("./MessageOverrideWhatsapp"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(a, n) { if (!(a instanceof n)) throw new TypeError("Cannot call a class as a function"); }
function _defineProperties(e, r) { for (var t = 0; t < r.length; t++) { var o = r[t]; o.enumerable = o.enumerable || !1, o.configurable = !0, "value" in o && (o.writable = !0), Object.defineProperty(e, _toPropertyKey(o.key), o); } }
function _createClass(e, r, t) { return r && _defineProperties(e.prototype, r), t && _defineProperties(e, t), Object.defineProperty(e, "prototype", { writable: !1 }), e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Sunshine Conversations API
 *
 * The version of the OpenAPI document: 16.0.1
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 *
 * OpenAPI Generator version: 4.3.1
 *
 * Do not edit the class manually.
 *
 */
/**
 * The MessageOverride model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/MessageOverride
 * @version 16.0.1
 */
var MessageOverride = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>MessageOverride</code>.
   * A raw payload containing a message that is sent directly to a channel. See [message overrides](https://developer.zendesk.com/documentation/conversations/messaging-platform/programmable-conversations/message-overrides/) for more information.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/MessageOverride
   * @implements module:sunshine-conversations-client/sunshine-conversations-client.model/MessageOverrideApple
   * @implements module:sunshine-conversations-client/sunshine-conversations-client.model/MessageOverrideLine
   * @implements module:sunshine-conversations-client/sunshine-conversations-client.model/MessageOverrideMessenger
   * @implements module:sunshine-conversations-client/sunshine-conversations-client.model/MessageOverrideWhatsapp
   */
  function MessageOverride() {
    _classCallCheck(this, MessageOverride);
    _MessageOverrideApple["default"].initialize(this);
    _MessageOverrideLine["default"].initialize(this);
    _MessageOverrideMessenger["default"].initialize(this);
    _MessageOverrideWhatsapp["default"].initialize(this);
    MessageOverride.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  return _createClass(MessageOverride, [{
    key: "getApple",
    value:
    /**
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/AppleMessageOverridePayload}
         */
    function getApple() {
      return this.apple;
    }

    /**
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/AppleMessageOverridePayload} apple
     */
  }, {
    key: "setApple",
    value: function setApple(apple) {
      this['apple'] = apple;
    }
    /**
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/MessageOverridePayload}
         */
  }, {
    key: "getLine",
    value: function getLine() {
      return this.line;
    }

    /**
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/MessageOverridePayload} line
     */
  }, {
    key: "setLine",
    value: function setLine(line) {
      this['line'] = line;
    }
    /**
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/MessageOverridePayload}
         */
  }, {
    key: "getMessenger",
    value: function getMessenger() {
      return this.messenger;
    }

    /**
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/MessageOverridePayload} messenger
     */
  }, {
    key: "setMessenger",
    value: function setMessenger(messenger) {
      this['messenger'] = messenger;
    }
    /**
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/MessageOverridePayload}
         */
  }, {
    key: "getWhatsapp",
    value: function getWhatsapp() {
      return this.whatsapp;
    }

    /**
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/MessageOverridePayload} whatsapp
     */
  }, {
    key: "setWhatsapp",
    value: function setWhatsapp(whatsapp) {
      this['whatsapp'] = whatsapp;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>MessageOverride</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/MessageOverride} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/MessageOverride} The populated <code>MessageOverride</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new MessageOverride();
        _MessageOverrideApple["default"].constructFromObject(data, obj);
        _MessageOverrideLine["default"].constructFromObject(data, obj);
        _MessageOverrideMessenger["default"].constructFromObject(data, obj);
        _MessageOverrideWhatsapp["default"].constructFromObject(data, obj);
        if (data.hasOwnProperty('apple')) {
          obj['apple'] = _AppleMessageOverridePayload["default"].constructFromObject(data['apple']);
        }
        if (data.hasOwnProperty('line')) {
          obj['line'] = _MessageOverridePayload["default"].constructFromObject(data['line']);
        }
        if (data.hasOwnProperty('messenger')) {
          obj['messenger'] = _MessageOverridePayload["default"].constructFromObject(data['messenger']);
        }
        if (data.hasOwnProperty('whatsapp')) {
          obj['whatsapp'] = _MessageOverridePayload["default"].constructFromObject(data['whatsapp']);
        }
      }
      return obj;
    }
  }]);
}();
/**
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/AppleMessageOverridePayload} apple
 */
MessageOverride.prototype['apple'] = undefined;

/**
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/MessageOverridePayload} line
 */
MessageOverride.prototype['line'] = undefined;

/**
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/MessageOverridePayload} messenger
 */
MessageOverride.prototype['messenger'] = undefined;

/**
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/MessageOverridePayload} whatsapp
 */
MessageOverride.prototype['whatsapp'] = undefined;

// Implement MessageOverrideApple interface:
/**
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/AppleMessageOverridePayload} apple
 */
_MessageOverrideApple["default"].prototype['apple'] = undefined;
// Implement MessageOverrideLine interface:
/**
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/MessageOverridePayload} line
 */
_MessageOverrideLine["default"].prototype['line'] = undefined;
// Implement MessageOverrideMessenger interface:
/**
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/MessageOverridePayload} messenger
 */
_MessageOverrideMessenger["default"].prototype['messenger'] = undefined;
// Implement MessageOverrideWhatsapp interface:
/**
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/MessageOverridePayload} whatsapp
 */
_MessageOverrideWhatsapp["default"].prototype['whatsapp'] = undefined;
var _default = exports["default"] = MessageOverride;
