// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/MetaConversionEventPayload.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(a, n) { if (!(a instanceof n)) throw new TypeError("Cannot call a class as a function"); }
function _defineProperties(e, r) { for (var t = 0; t < r.length; t++) { var o = r[t]; o.enumerable = o.enumerable || !1, o.configurable = !0, "value" in o && (o.writable = !0), Object.defineProperty(e, _toPropertyKey(o.key), o); } }
function _createClass(e, r, t) { return r && _defineProperties(e.prototype, r), t && _defineProperties(e, t), Object.defineProperty(e, "prototype", { writable: !1 }), e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Sunshine Conversations API
 *
 * The version of the OpenAPI document: 16.0.1
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 *
 * OpenAPI Generator version: 4.3.1
 *
 * Do not edit the class manually.
 *
 */
/**
 * The MetaConversionEventPayload model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/MetaConversionEventPayload
 * @version 16.0.1
 */
var MetaConversionEventPayload = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>MetaConversionEventPayload</code>.
   * The payload to be sent to Meta&#39;s conversion events API. Should contain the exact structure expected by [Meta&#39;s Conversions API](https://developers.facebook.com/docs/marketing-api/conversions-api/parameters/server-event) except for the &#x60;user_data&#x60; field, which will be provided by Zendesk.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/MetaConversionEventPayload
   * @param data {Array.<Object>} The array of conversion events to be sent to Meta's conversion events API.
   */
  function MetaConversionEventPayload(data) {
    _classCallCheck(this, MetaConversionEventPayload);
    MetaConversionEventPayload.initialize(this, data);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  return _createClass(MetaConversionEventPayload, [{
    key: "getData",
    value:
    /**
         * Returns The array of conversion events to be sent to Meta's conversion events API.
         * @return {Array.<Object>}
         */
    function getData() {
      return this.data;
    }

    /**
     * Sets The array of conversion events to be sent to Meta's conversion events API.
     * @param {Array.<Object>} data The array of conversion events to be sent to Meta's conversion events API.
     */
  }, {
    key: "setData",
    value: function setData(data) {
      this['data'] = data;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj, data) {
      obj['data'] = data;
    }

    /**
     * Constructs a <code>MetaConversionEventPayload</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/MetaConversionEventPayload} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/MetaConversionEventPayload} The populated <code>MetaConversionEventPayload</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new MetaConversionEventPayload();
        if (data.hasOwnProperty('data')) {
          obj['data'] = _ApiClient["default"].convertToType(data['data'], [Object]);
        }
      }
      return obj;
    }
  }]);
}();
/**
 * The array of conversion events to be sent to Meta's conversion events API.
 * @member {Array.<Object>} data
 */
MetaConversionEventPayload.prototype['data'] = undefined;
var _default = exports["default"] = MetaConversionEventPayload;
