// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/PassControlBody.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(a, n) { if (!(a instanceof n)) throw new TypeError("Cannot call a class as a function"); }
function _defineProperties(e, r) { for (var t = 0; t < r.length; t++) { var o = r[t]; o.enumerable = o.enumerable || !1, o.configurable = !0, "value" in o && (o.writable = !0), Object.defineProperty(e, _toPropertyKey(o.key), o); } }
function _createClass(e, r, t) { return r && _defineProperties(e.prototype, r), t && _defineProperties(e, t), Object.defineProperty(e, "prototype", { writable: !1 }), e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Sunshine Conversations API
 *
 * The version of the OpenAPI document: 16.0.1
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 *
 * OpenAPI Generator version: 4.3.1
 *
 * Do not edit the class manually.
 *
 */
/**
 * The PassControlBody model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/PassControlBody
 * @version 16.0.1
 */
var PassControlBody = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>PassControlBody</code>.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/PassControlBody
   * @param switchboardIntegration {String} The id or the name of the switchboard integration that will become active. May also use the `next` keyword to transfer control to the next switchboard integration designated in the switchboard hierarchy configuration
   */
  function PassControlBody(switchboardIntegration) {
    _classCallCheck(this, PassControlBody);
    PassControlBody.initialize(this, switchboardIntegration);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  return _createClass(PassControlBody, [{
    key: "getSwitchboardIntegration",
    value:
    /**
         * Returns The id or the name of the switchboard integration that will become active. May also use the `next` keyword to transfer control to the next switchboard integration designated in the switchboard hierarchy configuration
         * @return {String}
         */
    function getSwitchboardIntegration() {
      return this.switchboardIntegration;
    }

    /**
     * Sets The id or the name of the switchboard integration that will become active. May also use the `next` keyword to transfer control to the next switchboard integration designated in the switchboard hierarchy configuration
     * @param {String} switchboardIntegration The id or the name of the switchboard integration that will become active. May also use the `next` keyword to transfer control to the next switchboard integration designated in the switchboard hierarchy configuration
     */
  }, {
    key: "setSwitchboardIntegration",
    value: function setSwitchboardIntegration(switchboardIntegration) {
      this['switchboardIntegration'] = switchboardIntegration;
    }
    /**
         * Returns Reason for the pass control action.
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/PassControlBody.ReasonEnum}
         */
  }, {
    key: "getReason",
    value: function getReason() {
      return this.reason;
    }

    /**
     * Sets Reason for the pass control action.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/PassControlBody.ReasonEnum} reason Reason for the pass control action.
     */
  }, {
    key: "setReason",
    value: function setReason(reason) {
      this['reason'] = reason;
    }
    /**
         * Returns Flat object containing custom properties. Strings, numbers and booleans are the only supported format that can be passed to metadata. The metadata is limited to 4KB in size. The metadata object will be included in the `switchboard:passControl` webhook.
         * @return {Object}
         */
  }, {
    key: "getMetadata",
    value: function getMetadata() {
      return this.metadata;
    }

    /**
     * Sets Flat object containing custom properties. Strings, numbers and booleans are the only supported format that can be passed to metadata. The metadata is limited to 4KB in size. The metadata object will be included in the `switchboard:passControl` webhook.
     * @param {Object} metadata Flat object containing custom properties. Strings, numbers and booleans are the only supported format that can be passed to metadata. The metadata is limited to 4KB in size. The metadata object will be included in the `switchboard:passControl` webhook.
     */
  }, {
    key: "setMetadata",
    value: function setMetadata(metadata) {
      this['metadata'] = metadata;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj, switchboardIntegration) {
      obj['switchboardIntegration'] = switchboardIntegration;
    }

    /**
     * Constructs a <code>PassControlBody</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/PassControlBody} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/PassControlBody} The populated <code>PassControlBody</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new PassControlBody();
        if (data.hasOwnProperty('switchboardIntegration')) {
          obj['switchboardIntegration'] = _ApiClient["default"].convertToType(data['switchboardIntegration'], 'String');
        }
        if (data.hasOwnProperty('reason')) {
          obj['reason'] = _ApiClient["default"].convertToType(data['reason'], 'String');
        }
        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], Object);
        }
      }
      return obj;
    }
  }]);
}();
/**
 * The id or the name of the switchboard integration that will become active. May also use the `next` keyword to transfer control to the next switchboard integration designated in the switchboard hierarchy configuration
 * @member {String} switchboardIntegration
 */
PassControlBody.prototype['switchboardIntegration'] = undefined;

/**
 * Reason for the pass control action.
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/PassControlBody.ReasonEnum} reason
 */
PassControlBody.prototype['reason'] = undefined;

/**
 * Flat object containing custom properties. Strings, numbers and booleans are the only supported format that can be passed to metadata. The metadata is limited to 4KB in size. The metadata object will be included in the `switchboard:passControl` webhook.
 * @member {Object} metadata
 */
PassControlBody.prototype['metadata'] = undefined;

/**
 * Allowed values for the <code>reason</code> property.
 * @enum {String}
 * @readonly
 */
PassControlBody['ReasonEnum'] = {
  /**
   * value: "ticketClosed"
   * @const
   */
  "ticketClosed": "ticketClosed",
  /**
   * value: "transferToEmail"
   * @const
   */
  "transferToEmail": "transferToEmail"
};
var _default = exports["default"] = PassControlBody;
