// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/SyncUserBody.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(a, n) { if (!(a instanceof n)) throw new TypeError("Cannot call a class as a function"); }
function _defineProperties(e, r) { for (var t = 0; t < r.length; t++) { var o = r[t]; o.enumerable = o.enumerable || !1, o.configurable = !0, "value" in o && (o.writable = !0), Object.defineProperty(e, _toPropertyKey(o.key), o); } }
function _createClass(e, r, t) { return r && _defineProperties(e.prototype, r), t && _defineProperties(e, t), Object.defineProperty(e, "prototype", { writable: !1 }), e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Sunshine Conversations API
 *
 * The version of the OpenAPI document: 16.0.1
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 *
 * OpenAPI Generator version: 4.3.1
 *
 * Do not edit the class manually.
 *
 */
/**
 * The SyncUserBody model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/SyncUserBody
 * @version 16.0.1
 */
var SyncUserBody = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>SyncUserBody</code>.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/SyncUserBody
   */
  function SyncUserBody() {
    _classCallCheck(this, SyncUserBody);
    SyncUserBody.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  return _createClass(SyncUserBody, [{
    key: "getSurvivingZendeskId",
    value:
    /**
         * Returns Only used for synchronizing messaging users when their Support user counterparts have been merged. The user.id of the surviving Support user is specified here.  
         * @return {String}
         */
    function getSurvivingZendeskId() {
      return this.survivingZendeskId;
    }

    /**
     * Sets Only used for synchronizing messaging users when their Support user counterparts have been merged. The user.id of the surviving Support user is specified here.  
     * @param {String} survivingZendeskId Only used for synchronizing messaging users when their Support user counterparts have been merged. The user.id of the surviving Support user is specified here.  
     */
  }, {
    key: "setSurvivingZendeskId",
    value: function setSurvivingZendeskId(survivingZendeskId) {
      this['survivingZendeskId'] = survivingZendeskId;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>SyncUserBody</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/SyncUserBody} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/SyncUserBody} The populated <code>SyncUserBody</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new SyncUserBody();
        if (data.hasOwnProperty('survivingZendeskId')) {
          obj['survivingZendeskId'] = _ApiClient["default"].convertToType(data['survivingZendeskId'], 'String');
        }
      }
      return obj;
    }
  }]);
}();
/**
 * Only used for synchronizing messaging users when their Support user counterparts have been merged. The user.id of the surviving Support user is specified here.  
 * @member {String} survivingZendeskId
 */
SyncUserBody.prototype['survivingZendeskId'] = undefined;
var _default = exports["default"] = SyncUserBody;
