// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/TemplateMessage.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(a, n) { if (!(a instanceof n)) throw new TypeError("Cannot call a class as a function"); }
function _defineProperties(e, r) { for (var t = 0; t < r.length; t++) { var o = r[t]; o.enumerable = o.enumerable || !1, o.configurable = !0, "value" in o && (o.writable = !0), Object.defineProperty(e, _toPropertyKey(o.key), o); } }
function _createClass(e, r, t) { return r && _defineProperties(e.prototype, r), t && _defineProperties(e, t), Object.defineProperty(e, "prototype", { writable: !1 }), e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Sunshine Conversations API
 *
 * The version of the OpenAPI document: 16.0.1
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 *
 * OpenAPI Generator version: 4.3.1
 *
 * Do not edit the class manually.
 *
 */
/**
 * The TemplateMessage model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/TemplateMessage
 * @version 16.0.1
 */
var TemplateMessage = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>TemplateMessage</code>.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/TemplateMessage
   * @param type {String} The type of message.
   * @param template {Object} The whatsapp template message to send. For more information, consult the [guide](https://developer.zendesk.com/documentation/conversations/messaging-platform/programmable-conversations/message-overrides/#template-messages). `schema` must be set to `whatsapp`.
   */
  function TemplateMessage(type, template) {
    _classCallCheck(this, TemplateMessage);
    TemplateMessage.initialize(this, type, template);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  return _createClass(TemplateMessage, [{
    key: "getType",
    value:
    /**
         * Returns The type of message.
         * @return {String}
         */
    function getType() {
      return this.type;
    }

    /**
     * Sets The type of message.
     * @param {String} type The type of message.
     */
  }, {
    key: "setType",
    value: function setType(type) {
      this['type'] = type;
    }
    /**
         * Returns When set to true, the chat input will be disabled on supported client implementations when the message is the most recent one in the history. Can be used for guided flows or to temporarily disable the user's ability to send messages in the conversation.
         * @return {Boolean}
         */
  }, {
    key: "getBlockChatInput",
    value: function getBlockChatInput() {
      return this.blockChatInput;
    }

    /**
     * Sets When set to true, the chat input will be disabled on supported client implementations when the message is the most recent one in the history. Can be used for guided flows or to temporarily disable the user's ability to send messages in the conversation.
     * @param {Boolean} blockChatInput When set to true, the chat input will be disabled on supported client implementations when the message is the most recent one in the history. Can be used for guided flows or to temporarily disable the user's ability to send messages in the conversation.
     */
  }, {
    key: "setBlockChatInput",
    value: function setBlockChatInput(blockChatInput) {
      this['blockChatInput'] = blockChatInput;
    }
    /**
         * Returns The whatsapp template message to send. For more information, consult the [guide](https://developer.zendesk.com/documentation/conversations/messaging-platform/programmable-conversations/message-overrides/#template-messages). `schema` must be set to `whatsapp`.
         * @return {Object}
         */
  }, {
    key: "getTemplate",
    value: function getTemplate() {
      return this.template;
    }

    /**
     * Sets The whatsapp template message to send. For more information, consult the [guide](https://developer.zendesk.com/documentation/conversations/messaging-platform/programmable-conversations/message-overrides/#template-messages). `schema` must be set to `whatsapp`.
     * @param {Object} template The whatsapp template message to send. For more information, consult the [guide](https://developer.zendesk.com/documentation/conversations/messaging-platform/programmable-conversations/message-overrides/#template-messages). `schema` must be set to `whatsapp`.
     */
  }, {
    key: "setTemplate",
    value: function setTemplate(template) {
      this['template'] = template;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj, type, template) {
      obj['type'] = type;
      obj['template'] = template;
    }

    /**
     * Constructs a <code>TemplateMessage</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/TemplateMessage} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/TemplateMessage} The populated <code>TemplateMessage</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new TemplateMessage();
        if (data.hasOwnProperty('type')) {
          obj['type'] = _ApiClient["default"].convertToType(data['type'], 'String');
        }
        if (data.hasOwnProperty('blockChatInput')) {
          obj['blockChatInput'] = _ApiClient["default"].convertToType(data['blockChatInput'], 'Boolean');
        }
        if (data.hasOwnProperty('template')) {
          obj['template'] = _ApiClient["default"].convertToType(data['template'], Object);
        }
      }
      return obj;
    }
  }]);
}();
/**
 * The type of message.
 * @member {String} type
 * @default 'template'
 */
TemplateMessage.prototype['type'] = 'template';

/**
 * When set to true, the chat input will be disabled on supported client implementations when the message is the most recent one in the history. Can be used for guided flows or to temporarily disable the user's ability to send messages in the conversation.
 * @member {Boolean} blockChatInput
 */
TemplateMessage.prototype['blockChatInput'] = undefined;

/**
 * The whatsapp template message to send. For more information, consult the [guide](https://developer.zendesk.com/documentation/conversations/messaging-platform/programmable-conversations/message-overrides/#template-messages). `schema` must be set to `whatsapp`.
 * @member {Object} template
 */
TemplateMessage.prototype['template'] = undefined;
var _default = exports["default"] = TemplateMessage;
