// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/TransferToEmail.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(a, n) { if (!(a instanceof n)) throw new TypeError("Cannot call a class as a function"); }
function _defineProperties(e, r) { for (var t = 0; t < r.length; t++) { var o = r[t]; o.enumerable = o.enumerable || !1, o.configurable = !0, "value" in o && (o.writable = !0), Object.defineProperty(e, _toPropertyKey(o.key), o); } }
function _createClass(e, r, t) { return r && _defineProperties(e.prototype, r), t && _defineProperties(e, t), Object.defineProperty(e, "prototype", { writable: !1 }), e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Sunshine Conversations API
 *
 * The version of the OpenAPI document: 16.0.1
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 *
 * OpenAPI Generator version: 4.3.1
 *
 * Do not edit the class manually.
 *
 */
/**
 * The TransferToEmail model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/TransferToEmail
 * @version 16.0.1
 */
var TransferToEmail = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>TransferToEmail</code>.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/TransferToEmail
   * @param type {String} The type of system activity that generated the message. The value of this field determines the dynamic content that is rendered to the end-user / agent when viewing this message. Each `type` value will have a set of pre-defined, localized strings that describe the activity.
   */
  function TransferToEmail(type) {
    _classCallCheck(this, TransferToEmail);
    TransferToEmail.initialize(this, type);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  return _createClass(TransferToEmail, [{
    key: "getType",
    value:
    /**
         * Returns The type of system activity that generated the message. The value of this field determines the dynamic content that is rendered to the end-user / agent when viewing this message. Each `type` value will have a set of pre-defined, localized strings that describe the activity.
         * @return {String}
         */
    function getType() {
      return this.type;
    }

    /**
     * Sets The type of system activity that generated the message. The value of this field determines the dynamic content that is rendered to the end-user / agent when viewing this message. Each `type` value will have a set of pre-defined, localized strings that describe the activity.
     * @param {String} type The type of system activity that generated the message. The value of this field determines the dynamic content that is rendered to the end-user / agent when viewing this message. Each `type` value will have a set of pre-defined, localized strings that describe the activity.
     */
  }, {
    key: "setType",
    value: function setType(type) {
      this['type'] = type;
    }
    /**
         * Returns No additional data is supplied with the \"ticket:transfer:email\" activity type at this time.
         * @return {Object.<String, Object>}
         */
  }, {
    key: "getData",
    value: function getData() {
      return this.data;
    }

    /**
     * Sets No additional data is supplied with the \"ticket:transfer:email\" activity type at this time.
     * @param {Object.<String, Object>} data No additional data is supplied with the \"ticket:transfer:email\" activity type at this time.
     */
  }, {
    key: "setData",
    value: function setData(data) {
      this['data'] = data;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj, type) {
      obj['type'] = type;
    }

    /**
     * Constructs a <code>TransferToEmail</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/TransferToEmail} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/TransferToEmail} The populated <code>TransferToEmail</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new TransferToEmail();
        if (data.hasOwnProperty('type')) {
          obj['type'] = _ApiClient["default"].convertToType(data['type'], 'String');
        }
        if (data.hasOwnProperty('data')) {
          obj['data'] = _ApiClient["default"].convertToType(data['data'], {
            'String': Object
          });
        }
      }
      return obj;
    }
  }]);
}();
/**
 * The type of system activity that generated the message. The value of this field determines the dynamic content that is rendered to the end-user / agent when viewing this message. Each `type` value will have a set of pre-defined, localized strings that describe the activity.
 * @member {String} type
 * @default 'ticket:transfer:email'
 */
TransferToEmail.prototype['type'] = 'ticket:transfer:email';

/**
 * No additional data is supplied with the \"ticket:transfer:email\" activity type at this time.
 * @member {Object.<String, Object>} data
 */
TransferToEmail.prototype['data'] = undefined;
var _default = exports["default"] = TransferToEmail;
