// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/UnityUpdate.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _IntegrationUpdateBase = _interopRequireDefault(require("./IntegrationUpdateBase"));
var _UnityUpdateAllOf = _interopRequireDefault(require("./UnityUpdateAllOf"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(a, n) { if (!(a instanceof n)) throw new TypeError("Cannot call a class as a function"); }
function _defineProperties(e, r) { for (var t = 0; t < r.length; t++) { var o = r[t]; o.enumerable = o.enumerable || !1, o.configurable = !0, "value" in o && (o.writable = !0), Object.defineProperty(e, _toPropertyKey(o.key), o); } }
function _createClass(e, r, t) { return r && _defineProperties(e.prototype, r), t && _defineProperties(e, t), Object.defineProperty(e, "prototype", { writable: !1 }), e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Sunshine Conversations API
 *
 * The version of the OpenAPI document: 16.0.1
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 *
 * OpenAPI Generator version: 4.3.1
 *
 * Do not edit the class manually.
 *
 */
/**
 * The UnityUpdate model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/UnityUpdate
 * @version 16.0.1
 */
var UnityUpdate = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UnityUpdate</code>.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/UnityUpdate
   * @implements module:sunshine-conversations-client/sunshine-conversations-client.model/IntegrationUpdateBase
   * @implements module:sunshine-conversations-client/sunshine-conversations-client.model/UnityUpdateAllOf
   */
  function UnityUpdate() {
    _classCallCheck(this, UnityUpdate);
    _IntegrationUpdateBase["default"].initialize(this);
    _UnityUpdateAllOf["default"].initialize(this);
    UnityUpdate.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  return _createClass(UnityUpdate, [{
    key: "getDisplayName",
    value:
    /**
         * Returns A human-friendly name used to identify the integration. `displayName` can be unset by changing it to `null`.
         * @return {String}
         */
    function getDisplayName() {
      return this.displayName;
    }

    /**
     * Sets A human-friendly name used to identify the integration. `displayName` can be unset by changing it to `null`.
     * @param {String} displayName A human-friendly name used to identify the integration. `displayName` can be unset by changing it to `null`.
     */
  }, {
    key: "setDisplayName",
    value: function setDisplayName(displayName) {
      this['displayName'] = displayName;
    }
    /**
         * Returns The default responder ID for the integration. This is the ID of the responder that will be used to send messages to the user. For more information, refer to the <a href=\"https://developer.zendesk.com/documentation/conversations/messaging-platform/programmable-conversations/switchboard/#default-integration-assignment\">Switchboard guide</a>. 
         * @return {String}
         */
  }, {
    key: "getDefaultResponderId",
    value: function getDefaultResponderId() {
      return this.defaultResponderId;
    }

    /**
     * Sets The default responder ID for the integration. This is the ID of the responder that will be used to send messages to the user. For more information, refer to the <a href=\"https://developer.zendesk.com/documentation/conversations/messaging-platform/programmable-conversations/switchboard/#default-integration-assignment\">Switchboard guide</a>. 
     * @param {String} defaultResponderId The default responder ID for the integration. This is the ID of the responder that will be used to send messages to the user. For more information, refer to the <a href=\"https://developer.zendesk.com/documentation/conversations/messaging-platform/programmable-conversations/switchboard/#default-integration-assignment\">Switchboard guide</a>. 
     */
  }, {
    key: "setDefaultResponderId",
    value: function setDefaultResponderId(defaultResponderId) {
      this['defaultResponderId'] = defaultResponderId;
    }
    /**
         * Returns Allows users to view their list of conversations. By default, the list of conversations will be visible. *This setting only applies to apps where `settings.multiConvoEnabled` is set to `true`*. 
         * @return {Boolean}
         */
  }, {
    key: "getCanUserSeeConversationList",
    value: function getCanUserSeeConversationList() {
      return this.canUserSeeConversationList;
    }

    /**
     * Sets Allows users to view their list of conversations. By default, the list of conversations will be visible. *This setting only applies to apps where `settings.multiConvoEnabled` is set to `true`*. 
     * @param {Boolean} canUserSeeConversationList Allows users to view their list of conversations. By default, the list of conversations will be visible. *This setting only applies to apps where `settings.multiConvoEnabled` is set to `true`*. 
     */
  }, {
    key: "setCanUserSeeConversationList",
    value: function setCanUserSeeConversationList(canUserSeeConversationList) {
      this['canUserSeeConversationList'] = canUserSeeConversationList;
    }
    /**
         * Returns Allows users to create more than one conversation on the Unity integration.
         * @return {Boolean}
         */
  }, {
    key: "getCanUserCreateMoreConversations",
    value: function getCanUserCreateMoreConversations() {
      return this.canUserCreateMoreConversations;
    }

    /**
     * Sets Allows users to create more than one conversation on the Unity integration.
     * @param {Boolean} canUserCreateMoreConversations Allows users to create more than one conversation on the Unity integration.
     */
  }, {
    key: "setCanUserCreateMoreConversations",
    value: function setCanUserCreateMoreConversations(canUserCreateMoreConversations) {
      this['canUserCreateMoreConversations'] = canUserCreateMoreConversations;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>UnityUpdate</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/UnityUpdate} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/UnityUpdate} The populated <code>UnityUpdate</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UnityUpdate();
        _IntegrationUpdateBase["default"].constructFromObject(data, obj);
        _UnityUpdateAllOf["default"].constructFromObject(data, obj);
        if (data.hasOwnProperty('displayName')) {
          obj['displayName'] = _ApiClient["default"].convertToType(data['displayName'], 'String');
        }
        if (data.hasOwnProperty('defaultResponderId')) {
          obj['defaultResponderId'] = _ApiClient["default"].convertToType(data['defaultResponderId'], 'String');
        }
        if (data.hasOwnProperty('canUserSeeConversationList')) {
          obj['canUserSeeConversationList'] = _ApiClient["default"].convertToType(data['canUserSeeConversationList'], 'Boolean');
        }
        if (data.hasOwnProperty('canUserCreateMoreConversations')) {
          obj['canUserCreateMoreConversations'] = _ApiClient["default"].convertToType(data['canUserCreateMoreConversations'], 'Boolean');
        }
      }
      return obj;
    }
  }]);
}();
/**
 * A human-friendly name used to identify the integration. `displayName` can be unset by changing it to `null`.
 * @member {String} displayName
 */
UnityUpdate.prototype['displayName'] = undefined;

/**
 * The default responder ID for the integration. This is the ID of the responder that will be used to send messages to the user. For more information, refer to the <a href=\"https://developer.zendesk.com/documentation/conversations/messaging-platform/programmable-conversations/switchboard/#default-integration-assignment\">Switchboard guide</a>. 
 * @member {String} defaultResponderId
 */
UnityUpdate.prototype['defaultResponderId'] = undefined;

/**
 * Allows users to view their list of conversations. By default, the list of conversations will be visible. *This setting only applies to apps where `settings.multiConvoEnabled` is set to `true`*. 
 * @member {Boolean} canUserSeeConversationList
 */
UnityUpdate.prototype['canUserSeeConversationList'] = undefined;

/**
 * Allows users to create more than one conversation on the Unity integration.
 * @member {Boolean} canUserCreateMoreConversations
 */
UnityUpdate.prototype['canUserCreateMoreConversations'] = undefined;

// Implement IntegrationUpdateBase interface:
/**
 * A human-friendly name used to identify the integration. `displayName` can be unset by changing it to `null`.
 * @member {String} displayName
 */
_IntegrationUpdateBase["default"].prototype['displayName'] = undefined;
/**
 * The default responder ID for the integration. This is the ID of the responder that will be used to send messages to the user. For more information, refer to the <a href=\"https://developer.zendesk.com/documentation/conversations/messaging-platform/programmable-conversations/switchboard/#default-integration-assignment\">Switchboard guide</a>. 
 * @member {String} defaultResponderId
 */
_IntegrationUpdateBase["default"].prototype['defaultResponderId'] = undefined;
// Implement UnityUpdateAllOf interface:
/**
 * Allows users to view their list of conversations. By default, the list of conversations will be visible. *This setting only applies to apps where `settings.multiConvoEnabled` is set to `true`*. 
 * @member {Boolean} canUserSeeConversationList
 */
_UnityUpdateAllOf["default"].prototype['canUserSeeConversationList'] = undefined;
/**
 * Allows users to create more than one conversation on the Unity integration.
 * @member {Boolean} canUserCreateMoreConversations
 */
_UnityUpdateAllOf["default"].prototype['canUserCreateMoreConversations'] = undefined;
var _default = exports["default"] = UnityUpdate;
