// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/UserCreateBody.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _Profile = _interopRequireDefault(require("./Profile"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(a, n) { if (!(a instanceof n)) throw new TypeError("Cannot call a class as a function"); }
function _defineProperties(e, r) { for (var t = 0; t < r.length; t++) { var o = r[t]; o.enumerable = o.enumerable || !1, o.configurable = !0, "value" in o && (o.writable = !0), Object.defineProperty(e, _toPropertyKey(o.key), o); } }
function _createClass(e, r, t) { return r && _defineProperties(e.prototype, r), t && _defineProperties(e, t), Object.defineProperty(e, "prototype", { writable: !1 }), e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Sunshine Conversations API
 *
 * The version of the OpenAPI document: 16.0.1
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 *
 * OpenAPI Generator version: 4.3.1
 *
 * Do not edit the class manually.
 *
 */
/**
 * The UserCreateBody model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/UserCreateBody
 * @version 16.0.1
 */
var UserCreateBody = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UserCreateBody</code>.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/UserCreateBody
   * @param externalId {String} A unique identifier for the user. The `externalId` can be used to link a user to the same conversation [across multiple devices](https://developer.zendesk.com/documentation/conversations/messaging-platform/users/authenticating-users/). 
   */
  function UserCreateBody(externalId) {
    _classCallCheck(this, UserCreateBody);
    UserCreateBody.initialize(this, externalId);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  return _createClass(UserCreateBody, [{
    key: "getExternalId",
    value:
    /**
         * Returns A unique identifier for the user. The `externalId` can be used to link a user to the same conversation [across multiple devices](https://developer.zendesk.com/documentation/conversations/messaging-platform/users/authenticating-users/). 
         * @return {String}
         */
    function getExternalId() {
      return this.externalId;
    }

    /**
     * Sets A unique identifier for the user. The `externalId` can be used to link a user to the same conversation [across multiple devices](https://developer.zendesk.com/documentation/conversations/messaging-platform/users/authenticating-users/). 
     * @param {String} externalId A unique identifier for the user. The `externalId` can be used to link a user to the same conversation [across multiple devices](https://developer.zendesk.com/documentation/conversations/messaging-platform/users/authenticating-users/). 
     */
  }, {
    key: "setExternalId",
    value: function setExternalId(externalId) {
      this['externalId'] = externalId;
    }
    /**
         * Returns The date at which the user signed up. Must be ISO 8601 time format `YYYY-MM-DDThh:mm:ss.sssZ`.
         * @return {String}
         */
  }, {
    key: "getSignedUpAt",
    value: function getSignedUpAt() {
      return this.signedUpAt;
    }

    /**
     * Sets The date at which the user signed up. Must be ISO 8601 time format `YYYY-MM-DDThh:mm:ss.sssZ`.
     * @param {String} signedUpAt The date at which the user signed up. Must be ISO 8601 time format `YYYY-MM-DDThh:mm:ss.sssZ`.
     */
  }, {
    key: "setSignedUpAt",
    value: function setSignedUpAt(signedUpAt) {
      this['signedUpAt'] = signedUpAt;
    }
    /**
         * Returns Flag indicating whether a user should be retained after they have passed their inactive expiry. See [creating deletion schedules for bot-only conversations](https://support.zendesk.com/hc/en-us/articles/8499219792154) for more information.
         * @return {Boolean}
         */
  }, {
    key: "getToBeRetained",
    value: function getToBeRetained() {
      return this.toBeRetained;
    }

    /**
     * Sets Flag indicating whether a user should be retained after they have passed their inactive expiry. See [creating deletion schedules for bot-only conversations](https://support.zendesk.com/hc/en-us/articles/8499219792154) for more information.
     * @param {Boolean} toBeRetained Flag indicating whether a user should be retained after they have passed their inactive expiry. See [creating deletion schedules for bot-only conversations](https://support.zendesk.com/hc/en-us/articles/8499219792154) for more information.
     */
  }, {
    key: "setToBeRetained",
    value: function setToBeRetained(toBeRetained) {
      this['toBeRetained'] = toBeRetained;
    }
    /**
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/Profile}
         */
  }, {
    key: "getProfile",
    value: function getProfile() {
      return this.profile;
    }

    /**
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/Profile} profile
     */
  }, {
    key: "setProfile",
    value: function setProfile(profile) {
      this['profile'] = profile;
    }
    /**
         * Returns Flat object containing custom properties. Strings, numbers and booleans  are the only supported format that can be passed to metadata. The metadata is limited to 4KB in size. 
         * @return {Object.<String, Object>}
         */
  }, {
    key: "getMetadata",
    value: function getMetadata() {
      return this.metadata;
    }

    /**
     * Sets Flat object containing custom properties. Strings, numbers and booleans  are the only supported format that can be passed to metadata. The metadata is limited to 4KB in size. 
     * @param {Object.<String, Object>} metadata Flat object containing custom properties. Strings, numbers and booleans  are the only supported format that can be passed to metadata. The metadata is limited to 4KB in size. 
     */
  }, {
    key: "setMetadata",
    value: function setMetadata(metadata) {
      this['metadata'] = metadata;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj, externalId) {
      obj['externalId'] = externalId;
    }

    /**
     * Constructs a <code>UserCreateBody</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/UserCreateBody} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/UserCreateBody} The populated <code>UserCreateBody</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UserCreateBody();
        if (data.hasOwnProperty('externalId')) {
          obj['externalId'] = _ApiClient["default"].convertToType(data['externalId'], 'String');
        }
        if (data.hasOwnProperty('signedUpAt')) {
          obj['signedUpAt'] = _ApiClient["default"].convertToType(data['signedUpAt'], 'String');
        }
        if (data.hasOwnProperty('toBeRetained')) {
          obj['toBeRetained'] = _ApiClient["default"].convertToType(data['toBeRetained'], 'Boolean');
        }
        if (data.hasOwnProperty('profile')) {
          obj['profile'] = _Profile["default"].constructFromObject(data['profile']);
        }
        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], {
            'String': Object
          });
        }
      }
      return obj;
    }
  }]);
}();
/**
 * A unique identifier for the user. The `externalId` can be used to link a user to the same conversation [across multiple devices](https://developer.zendesk.com/documentation/conversations/messaging-platform/users/authenticating-users/). 
 * @member {String} externalId
 */
UserCreateBody.prototype['externalId'] = undefined;

/**
 * The date at which the user signed up. Must be ISO 8601 time format `YYYY-MM-DDThh:mm:ss.sssZ`.
 * @member {String} signedUpAt
 */
UserCreateBody.prototype['signedUpAt'] = undefined;

/**
 * Flag indicating whether a user should be retained after they have passed their inactive expiry. See [creating deletion schedules for bot-only conversations](https://support.zendesk.com/hc/en-us/articles/8499219792154) for more information.
 * @member {Boolean} toBeRetained
 */
UserCreateBody.prototype['toBeRetained'] = undefined;

/**
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/Profile} profile
 */
UserCreateBody.prototype['profile'] = undefined;

/**
 * Flat object containing custom properties. Strings, numbers and booleans  are the only supported format that can be passed to metadata. The metadata is limited to 4KB in size. 
 * @member {Object.<String, Object>} metadata
 */
UserCreateBody.prototype['metadata'] = undefined;
var _default = exports["default"] = UserCreateBody;
