// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/UserTruncated.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(a, n) { if (!(a instanceof n)) throw new TypeError("Cannot call a class as a function"); }
function _defineProperties(e, r) { for (var t = 0; t < r.length; t++) { var o = r[t]; o.enumerable = o.enumerable || !1, o.configurable = !0, "value" in o && (o.writable = !0), Object.defineProperty(e, _toPropertyKey(o.key), o); } }
function _createClass(e, r, t) { return r && _defineProperties(e.prototype, r), t && _defineProperties(e, t), Object.defineProperty(e, "prototype", { writable: !1 }), e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Sunshine Conversations API
 *
 * The version of the OpenAPI document: 16.0.1
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 *
 * OpenAPI Generator version: 4.3.1
 *
 * Do not edit the class manually.
 *
 */
/**
 * The UserTruncated model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/UserTruncated
 * @version 16.0.1
 */
var UserTruncated = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UserTruncated</code>.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/UserTruncated
   */
  function UserTruncated() {
    _classCallCheck(this, UserTruncated);
    UserTruncated.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  return _createClass(UserTruncated, [{
    key: "getId",
    value:
    /**
         * Returns The unique ID of the user.
         * @return {String}
         */
    function getId() {
      return this.id;
    }

    /**
     * Sets The unique ID of the user.
     * @param {String} id The unique ID of the user.
     */
  }, {
    key: "setId",
    value: function setId(id) {
      this['id'] = id;
    }
    /**
         * Returns Whether or not the user has authenticated, either via JWT or via the Help Center
         * @return {Boolean}
         */
  }, {
    key: "getAuthenticated",
    value: function getAuthenticated() {
      return this.authenticated;
    }

    /**
     * Sets Whether or not the user has authenticated, either via JWT or via the Help Center
     * @param {Boolean} authenticated Whether or not the user has authenticated, either via JWT or via the Help Center
     */
  }, {
    key: "setAuthenticated",
    value: function setAuthenticated(authenticated) {
      this['authenticated'] = authenticated;
    }
    /**
         * Returns An optional ID that can also be used to retrieve the user. 
         * @return {String}
         */
  }, {
    key: "getExternalId",
    value: function getExternalId() {
      return this.externalId;
    }

    /**
     * Sets An optional ID that can also be used to retrieve the user. 
     * @param {String} externalId An optional ID that can also be used to retrieve the user. 
     */
  }, {
    key: "setExternalId",
    value: function setExternalId(externalId) {
      this['externalId'] = externalId;
    }
    /**
         * Returns The ID that links a messaging user to its core Zendesk user counterpart. This ID can be used to fetch the core user record via the Zendesk Support API. 
         * @return {String}
         */
  }, {
    key: "getZendeskId",
    value: function getZendeskId() {
      return this.zendeskId;
    }

    /**
     * Sets The ID that links a messaging user to its core Zendesk user counterpart. This ID can be used to fetch the core user record via the Zendesk Support API. 
     * @param {String} zendeskId The ID that links a messaging user to its core Zendesk user counterpart. This ID can be used to fetch the core user record via the Zendesk Support API. 
     */
  }, {
    key: "setZendeskId",
    value: function setZendeskId(zendeskId) {
      this['zendeskId'] = zendeskId;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>UserTruncated</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/UserTruncated} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/UserTruncated} The populated <code>UserTruncated</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UserTruncated();
        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }
        if (data.hasOwnProperty('authenticated')) {
          obj['authenticated'] = _ApiClient["default"].convertToType(data['authenticated'], 'Boolean');
        }
        if (data.hasOwnProperty('externalId')) {
          obj['externalId'] = _ApiClient["default"].convertToType(data['externalId'], 'String');
        }
        if (data.hasOwnProperty('zendeskId')) {
          obj['zendeskId'] = _ApiClient["default"].convertToType(data['zendeskId'], 'String');
        }
      }
      return obj;
    }
  }]);
}();
/**
 * The unique ID of the user.
 * @member {String} id
 */
UserTruncated.prototype['id'] = undefined;

/**
 * Whether or not the user has authenticated, either via JWT or via the Help Center
 * @member {Boolean} authenticated
 */
UserTruncated.prototype['authenticated'] = undefined;

/**
 * An optional ID that can also be used to retrieve the user. 
 * @member {String} externalId
 */
UserTruncated.prototype['externalId'] = undefined;

/**
 * The ID that links a messaging user to its core Zendesk user counterpart. This ID can be used to fetch the core user record via the Zendesk Support API. 
 * @member {String} zendeskId
 */
UserTruncated.prototype['zendeskId'] = undefined;
var _default = exports["default"] = UserTruncated;
