// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/Viber.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _DefaultResponder = _interopRequireDefault(require("./DefaultResponder"));
var _DefaultResponderDefaultResponder = _interopRequireDefault(require("./DefaultResponderDefaultResponder"));
var _DefaultResponderId = _interopRequireDefault(require("./DefaultResponderId"));
var _Integration = _interopRequireDefault(require("./Integration"));
var _Status = _interopRequireDefault(require("./Status"));
var _ViberAllOf = _interopRequireDefault(require("./ViberAllOf"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(a, n) { if (!(a instanceof n)) throw new TypeError("Cannot call a class as a function"); }
function _defineProperties(e, r) { for (var t = 0; t < r.length; t++) { var o = r[t]; o.enumerable = o.enumerable || !1, o.configurable = !0, "value" in o && (o.writable = !0), Object.defineProperty(e, _toPropertyKey(o.key), o); } }
function _createClass(e, r, t) { return r && _defineProperties(e.prototype, r), t && _defineProperties(e, t), Object.defineProperty(e, "prototype", { writable: !1 }), e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Sunshine Conversations API
 *
 * The version of the OpenAPI document: 16.0.1
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 *
 * OpenAPI Generator version: 4.3.1
 *
 * Do not edit the class manually.
 *
 */
/**
 * The Viber model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/Viber
 * @version 16.0.1
 */
var Viber = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Viber</code>.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/Viber
   * @extends module:sunshine-conversations-client/sunshine-conversations-client.model/Integration
   * @implements module:sunshine-conversations-client/sunshine-conversations-client.model/Integration
   * @implements module:sunshine-conversations-client/sunshine-conversations-client.model/ViberAllOf
   * @implements module:sunshine-conversations-client/sunshine-conversations-client.model/DefaultResponderId
   * @implements module:sunshine-conversations-client/sunshine-conversations-client.model/DefaultResponder
   * @param type {String} To configure a Viber integration, acquire the Viber Public Account token from the user and call the Create Integration endpoint. 
   * @param token {String} Viber Public Account token.
   */
  function Viber(type, token) {
    _classCallCheck(this, Viber);
    _Integration["default"].initialize(this, type);
    _ViberAllOf["default"].initialize(this, token);
    _DefaultResponderId["default"].initialize(this);
    _DefaultResponder["default"].initialize(this);
    Viber.initialize(this, type, token);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  return _createClass(Viber, [{
    key: "getType",
    value:
    /**
         * Returns To configure a Viber integration, acquire the Viber Public Account token from the user and call the Create Integration endpoint. 
         * @return {String}
         */
    function getType() {
      return this.type;
    }

    /**
     * Sets To configure a Viber integration, acquire the Viber Public Account token from the user and call the Create Integration endpoint. 
     * @param {String} type To configure a Viber integration, acquire the Viber Public Account token from the user and call the Create Integration endpoint. 
     */
  }, {
    key: "setType",
    value: function setType(type) {
      this['type'] = type;
    }
    /**
         * Returns Viber Public Account token.
         * @return {String}
         */
  }, {
    key: "getToken",
    value: function getToken() {
      return this.token;
    }

    /**
     * Sets Viber Public Account token.
     * @param {String} token Viber Public Account token.
     */
  }, {
    key: "setToken",
    value: function setToken(token) {
      this['token'] = token;
    }
    /**
         * Returns Unique URI of the Viber account.
         * @return {String}
         */
  }, {
    key: "getUri",
    value: function getUri() {
      return this.uri;
    }

    /**
     * Sets Unique URI of the Viber account.
     * @param {String} uri Unique URI of the Viber account.
     */
  }, {
    key: "setUri",
    value: function setUri(uri) {
      this['uri'] = uri;
    }
    /**
         * Returns Unique ID of the Viber account.
         * @return {String}
         */
  }, {
    key: "getAccountId",
    value: function getAccountId() {
      return this.accountId;
    }

    /**
     * Sets Unique ID of the Viber account.
     * @param {String} accountId Unique ID of the Viber account.
     */
  }, {
    key: "setAccountId",
    value: function setAccountId(accountId) {
      this['accountId'] = accountId;
    }
    /**
         * Returns The default responder ID for the integration. This is the ID of the responder that will be used to send messages to the user. For more information, refer to the <a href=\"https://developer.zendesk.com/documentation/conversations/messaging-platform/programmable-conversations/switchboard/#default-integration-assignment\">Switchboard guide</a>. 
         * @return {String}
         */
  }, {
    key: "getDefaultResponderId",
    value: function getDefaultResponderId() {
      return this.defaultResponderId;
    }

    /**
     * Sets The default responder ID for the integration. This is the ID of the responder that will be used to send messages to the user. For more information, refer to the <a href=\"https://developer.zendesk.com/documentation/conversations/messaging-platform/programmable-conversations/switchboard/#default-integration-assignment\">Switchboard guide</a>. 
     * @param {String} defaultResponderId The default responder ID for the integration. This is the ID of the responder that will be used to send messages to the user. For more information, refer to the <a href=\"https://developer.zendesk.com/documentation/conversations/messaging-platform/programmable-conversations/switchboard/#default-integration-assignment\">Switchboard guide</a>. 
     */
  }, {
    key: "setDefaultResponderId",
    value: function setDefaultResponderId(defaultResponderId) {
      this['defaultResponderId'] = defaultResponderId;
    }
    /**
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/DefaultResponderDefaultResponder}
         */
  }, {
    key: "getDefaultResponder",
    value: function getDefaultResponder() {
      return this.defaultResponder;
    }

    /**
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/DefaultResponderDefaultResponder} defaultResponder
     */
  }, {
    key: "setDefaultResponder",
    value: function setDefaultResponder(defaultResponder) {
      this['defaultResponder'] = defaultResponder;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj, type, token) {
      obj['token'] = token;
    }

    /**
     * Constructs a <code>Viber</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/Viber} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/Viber} The populated <code>Viber</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Viber();
        _Integration["default"].constructFromObject(data, obj);
        _Integration["default"].constructFromObject(data, obj);
        _ViberAllOf["default"].constructFromObject(data, obj);
        _DefaultResponderId["default"].constructFromObject(data, obj);
        _DefaultResponder["default"].constructFromObject(data, obj);
        if (data.hasOwnProperty('type')) {
          obj['type'] = _ApiClient["default"].convertToType(data['type'], 'String');
        }
        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }
        if (data.hasOwnProperty('uri')) {
          obj['uri'] = _ApiClient["default"].convertToType(data['uri'], 'String');
        }
        if (data.hasOwnProperty('accountId')) {
          obj['accountId'] = _ApiClient["default"].convertToType(data['accountId'], 'String');
        }
        if (data.hasOwnProperty('defaultResponderId')) {
          obj['defaultResponderId'] = _ApiClient["default"].convertToType(data['defaultResponderId'], 'String');
        }
        if (data.hasOwnProperty('defaultResponder')) {
          obj['defaultResponder'] = _DefaultResponderDefaultResponder["default"].constructFromObject(data['defaultResponder']);
        }
      }
      return obj;
    }
  }]);
}();
/**
 * To configure a Viber integration, acquire the Viber Public Account token from the user and call the Create Integration endpoint. 
 * @member {String} type
 * @default 'viber'
 */
Viber.prototype['type'] = 'viber';

/**
 * Viber Public Account token.
 * @member {String} token
 */
Viber.prototype['token'] = undefined;

/**
 * Unique URI of the Viber account.
 * @member {String} uri
 */
Viber.prototype['uri'] = undefined;

/**
 * Unique ID of the Viber account.
 * @member {String} accountId
 */
Viber.prototype['accountId'] = undefined;

/**
 * The default responder ID for the integration. This is the ID of the responder that will be used to send messages to the user. For more information, refer to the <a href=\"https://developer.zendesk.com/documentation/conversations/messaging-platform/programmable-conversations/switchboard/#default-integration-assignment\">Switchboard guide</a>. 
 * @member {String} defaultResponderId
 */
Viber.prototype['defaultResponderId'] = undefined;

/**
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/DefaultResponderDefaultResponder} defaultResponder
 */
Viber.prototype['defaultResponder'] = undefined;

// Implement Integration interface:
/**
 * The unique ID of the integration.
 * @member {String} id
 */
_Integration["default"].prototype['id'] = undefined;
/**
 * @member {String} type
 */
_Integration["default"].prototype['type'] = undefined;
/**
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/Status} status
 */
_Integration["default"].prototype['status'] = undefined;
/**
 * A human-friendly name used to identify the integration.
 * @member {String} displayName
 */
_Integration["default"].prototype['displayName'] = undefined;
// Implement ViberAllOf interface:
/**
 * To configure a Viber integration, acquire the Viber Public Account token from the user and call the Create Integration endpoint. 
 * @member {String} type
 * @default 'viber'
 */
_ViberAllOf["default"].prototype['type'] = 'viber';
/**
 * Viber Public Account token.
 * @member {String} token
 */
_ViberAllOf["default"].prototype['token'] = undefined;
/**
 * Unique URI of the Viber account.
 * @member {String} uri
 */
_ViberAllOf["default"].prototype['uri'] = undefined;
/**
 * Unique ID of the Viber account.
 * @member {String} accountId
 */
_ViberAllOf["default"].prototype['accountId'] = undefined;
// Implement DefaultResponderId interface:
/**
 * The default responder ID for the integration. This is the ID of the responder that will be used to send messages to the user. For more information, refer to the <a href=\"https://developer.zendesk.com/documentation/conversations/messaging-platform/programmable-conversations/switchboard/#default-integration-assignment\">Switchboard guide</a>. 
 * @member {String} defaultResponderId
 */
_DefaultResponderId["default"].prototype['defaultResponderId'] = undefined;
// Implement DefaultResponder interface:
/**
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/DefaultResponderDefaultResponder} defaultResponder
 */
_DefaultResponder["default"].prototype['defaultResponder'] = undefined;
var _default = exports["default"] = Viber;
