// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/Whatsapp.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _DefaultResponder = _interopRequireDefault(require("./DefaultResponder"));
var _DefaultResponderDefaultResponder = _interopRequireDefault(require("./DefaultResponderDefaultResponder"));
var _DefaultResponderId = _interopRequireDefault(require("./DefaultResponderId"));
var _Integration = _interopRequireDefault(require("./Integration"));
var _Status = _interopRequireDefault(require("./Status"));
var _WhatsappAllOf = _interopRequireDefault(require("./WhatsappAllOf"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(a, n) { if (!(a instanceof n)) throw new TypeError("Cannot call a class as a function"); }
function _defineProperties(e, r) { for (var t = 0; t < r.length; t++) { var o = r[t]; o.enumerable = o.enumerable || !1, o.configurable = !0, "value" in o && (o.writable = !0), Object.defineProperty(e, _toPropertyKey(o.key), o); } }
function _createClass(e, r, t) { return r && _defineProperties(e.prototype, r), t && _defineProperties(e, t), Object.defineProperty(e, "prototype", { writable: !1 }), e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Sunshine Conversations API
 *
 * The version of the OpenAPI document: 16.0.1
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 *
 * OpenAPI Generator version: 4.3.1
 *
 * Do not edit the class manually.
 *
 */
/**
 * The Whatsapp model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/Whatsapp
 * @version 16.0.1
 */
var Whatsapp = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Whatsapp</code>.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/Whatsapp
   * @extends module:sunshine-conversations-client/sunshine-conversations-client.model/Integration
   * @implements module:sunshine-conversations-client/sunshine-conversations-client.model/Integration
   * @implements module:sunshine-conversations-client/sunshine-conversations-client.model/WhatsappAllOf
   * @implements module:sunshine-conversations-client/sunshine-conversations-client.model/DefaultResponderId
   * @implements module:sunshine-conversations-client/sunshine-conversations-client.model/DefaultResponder
   * @param type {String} To configure a WhatsApp integration, use your WhatsApp API Client connection information. Sunshine Conversations can provide WhatsApp API Client hosting for approved brands. See our [WhatsApp guide](https://docs.smooch.io/guide/whatsapp/#whatsapp-api-client) for more details on WhatsApp API Client hosting. 
   */
  function Whatsapp(type) {
    _classCallCheck(this, Whatsapp);
    _Integration["default"].initialize(this, type);
    _WhatsappAllOf["default"].initialize(this);
    _DefaultResponderId["default"].initialize(this);
    _DefaultResponder["default"].initialize(this);
    Whatsapp.initialize(this, type);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  return _createClass(Whatsapp, [{
    key: "getType",
    value:
    /**
         * Returns To configure a WhatsApp integration, use your WhatsApp API Client connection information. Sunshine Conversations can provide WhatsApp API Client hosting for approved brands. See our [WhatsApp guide](https://docs.smooch.io/guide/whatsapp/#whatsapp-api-client) for more details on WhatsApp API Client hosting. 
         * @return {String}
         */
    function getType() {
      return this.type;
    }

    /**
     * Sets To configure a WhatsApp integration, use your WhatsApp API Client connection information. Sunshine Conversations can provide WhatsApp API Client hosting for approved brands. See our [WhatsApp guide](https://docs.smooch.io/guide/whatsapp/#whatsapp-api-client) for more details on WhatsApp API Client hosting. 
     * @param {String} type To configure a WhatsApp integration, use your WhatsApp API Client connection information. Sunshine Conversations can provide WhatsApp API Client hosting for approved brands. See our [WhatsApp guide](https://docs.smooch.io/guide/whatsapp/#whatsapp-api-client) for more details on WhatsApp API Client hosting. 
     */
  }, {
    key: "setType",
    value: function setType(type) {
      this['type'] = type;
    }
    /**
         * Returns Specify a fallback language to use when sending WhatsApp message template using the short hand syntax. Defaults to en_US. See WhatsApp documentation for more info.
         * @return {String}
         */
  }, {
    key: "getHsmFallbackLanguage",
    value: function getHsmFallbackLanguage() {
      return this.hsmFallbackLanguage;
    }

    /**
     * Sets Specify a fallback language to use when sending WhatsApp message template using the short hand syntax. Defaults to en_US. See WhatsApp documentation for more info.
     * @param {String} hsmFallbackLanguage Specify a fallback language to use when sending WhatsApp message template using the short hand syntax. Defaults to en_US. See WhatsApp documentation for more info.
     */
  }, {
    key: "setHsmFallbackLanguage",
    value: function setHsmFallbackLanguage(hsmFallbackLanguage) {
      this['hsmFallbackLanguage'] = hsmFallbackLanguage;
    }
    /**
         * Returns The business ID associated with the WhatsApp account. In combination with accountManagementAccessToken, it’s used for Message Template Reconstruction.
         * @return {String}
         */
  }, {
    key: "getAccountId",
    value: function getAccountId() {
      return this.accountId;
    }

    /**
     * Sets The business ID associated with the WhatsApp account. In combination with accountManagementAccessToken, it’s used for Message Template Reconstruction.
     * @param {String} accountId The business ID associated with the WhatsApp account. In combination with accountManagementAccessToken, it’s used for Message Template Reconstruction.
     */
  }, {
    key: "setAccountId",
    value: function setAccountId(accountId) {
      this['accountId'] = accountId;
    }
    /**
         * Returns An access token associated with the accountId used to query the WhatsApp Account Management API. In combination with accountId, it’s used for Message Template Reconstruction.
         * @return {String}
         */
  }, {
    key: "getAccountManagementAccessToken",
    value: function getAccountManagementAccessToken() {
      return this.accountManagementAccessToken;
    }

    /**
     * Sets An access token associated with the accountId used to query the WhatsApp Account Management API. In combination with accountId, it’s used for Message Template Reconstruction.
     * @param {String} accountManagementAccessToken An access token associated with the accountId used to query the WhatsApp Account Management API. In combination with accountId, it’s used for Message Template Reconstruction.
     */
  }, {
    key: "setAccountManagementAccessToken",
    value: function setAccountManagementAccessToken(accountManagementAccessToken) {
      this['accountManagementAccessToken'] = accountManagementAccessToken;
    }
    /**
         * Returns The phone number that is associated with the deployment of this integration, if one exists.
         * @return {String}
         */
  }, {
    key: "getPhoneNumber",
    value: function getPhoneNumber() {
      return this.phoneNumber;
    }

    /**
     * Sets The phone number that is associated with the deployment of this integration, if one exists.
     * @param {String} phoneNumber The phone number that is associated with the deployment of this integration, if one exists.
     */
  }, {
    key: "setPhoneNumber",
    value: function setPhoneNumber(phoneNumber) {
      this['phoneNumber'] = phoneNumber;
    }
    /**
         * Returns The default responder ID for the integration. This is the ID of the responder that will be used to send messages to the user. For more information, refer to the <a href=\"https://developer.zendesk.com/documentation/conversations/messaging-platform/programmable-conversations/switchboard/#default-integration-assignment\">Switchboard guide</a>. 
         * @return {String}
         */
  }, {
    key: "getDefaultResponderId",
    value: function getDefaultResponderId() {
      return this.defaultResponderId;
    }

    /**
     * Sets The default responder ID for the integration. This is the ID of the responder that will be used to send messages to the user. For more information, refer to the <a href=\"https://developer.zendesk.com/documentation/conversations/messaging-platform/programmable-conversations/switchboard/#default-integration-assignment\">Switchboard guide</a>. 
     * @param {String} defaultResponderId The default responder ID for the integration. This is the ID of the responder that will be used to send messages to the user. For more information, refer to the <a href=\"https://developer.zendesk.com/documentation/conversations/messaging-platform/programmable-conversations/switchboard/#default-integration-assignment\">Switchboard guide</a>. 
     */
  }, {
    key: "setDefaultResponderId",
    value: function setDefaultResponderId(defaultResponderId) {
      this['defaultResponderId'] = defaultResponderId;
    }
    /**
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/DefaultResponderDefaultResponder}
         */
  }, {
    key: "getDefaultResponder",
    value: function getDefaultResponder() {
      return this.defaultResponder;
    }

    /**
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/DefaultResponderDefaultResponder} defaultResponder
     */
  }, {
    key: "setDefaultResponder",
    value: function setDefaultResponder(defaultResponder) {
      this['defaultResponder'] = defaultResponder;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj, type) {}

    /**
     * Constructs a <code>Whatsapp</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/Whatsapp} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/Whatsapp} The populated <code>Whatsapp</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Whatsapp();
        _Integration["default"].constructFromObject(data, obj);
        _Integration["default"].constructFromObject(data, obj);
        _WhatsappAllOf["default"].constructFromObject(data, obj);
        _DefaultResponderId["default"].constructFromObject(data, obj);
        _DefaultResponder["default"].constructFromObject(data, obj);
        if (data.hasOwnProperty('type')) {
          obj['type'] = _ApiClient["default"].convertToType(data['type'], 'String');
        }
        if (data.hasOwnProperty('hsmFallbackLanguage')) {
          obj['hsmFallbackLanguage'] = _ApiClient["default"].convertToType(data['hsmFallbackLanguage'], 'String');
        }
        if (data.hasOwnProperty('accountId')) {
          obj['accountId'] = _ApiClient["default"].convertToType(data['accountId'], 'String');
        }
        if (data.hasOwnProperty('accountManagementAccessToken')) {
          obj['accountManagementAccessToken'] = _ApiClient["default"].convertToType(data['accountManagementAccessToken'], 'String');
        }
        if (data.hasOwnProperty('phoneNumber')) {
          obj['phoneNumber'] = _ApiClient["default"].convertToType(data['phoneNumber'], 'String');
        }
        if (data.hasOwnProperty('defaultResponderId')) {
          obj['defaultResponderId'] = _ApiClient["default"].convertToType(data['defaultResponderId'], 'String');
        }
        if (data.hasOwnProperty('defaultResponder')) {
          obj['defaultResponder'] = _DefaultResponderDefaultResponder["default"].constructFromObject(data['defaultResponder']);
        }
      }
      return obj;
    }
  }]);
}();
/**
 * To configure a WhatsApp integration, use your WhatsApp API Client connection information. Sunshine Conversations can provide WhatsApp API Client hosting for approved brands. See our [WhatsApp guide](https://docs.smooch.io/guide/whatsapp/#whatsapp-api-client) for more details on WhatsApp API Client hosting. 
 * @member {String} type
 * @default 'whatsapp'
 */
Whatsapp.prototype['type'] = 'whatsapp';

/**
 * Specify a fallback language to use when sending WhatsApp message template using the short hand syntax. Defaults to en_US. See WhatsApp documentation for more info.
 * @member {String} hsmFallbackLanguage
 * @default 'en_US'
 */
Whatsapp.prototype['hsmFallbackLanguage'] = 'en_US';

/**
 * The business ID associated with the WhatsApp account. In combination with accountManagementAccessToken, it’s used for Message Template Reconstruction.
 * @member {String} accountId
 */
Whatsapp.prototype['accountId'] = undefined;

/**
 * An access token associated with the accountId used to query the WhatsApp Account Management API. In combination with accountId, it’s used for Message Template Reconstruction.
 * @member {String} accountManagementAccessToken
 */
Whatsapp.prototype['accountManagementAccessToken'] = undefined;

/**
 * The phone number that is associated with the deployment of this integration, if one exists.
 * @member {String} phoneNumber
 */
Whatsapp.prototype['phoneNumber'] = undefined;

/**
 * The default responder ID for the integration. This is the ID of the responder that will be used to send messages to the user. For more information, refer to the <a href=\"https://developer.zendesk.com/documentation/conversations/messaging-platform/programmable-conversations/switchboard/#default-integration-assignment\">Switchboard guide</a>. 
 * @member {String} defaultResponderId
 */
Whatsapp.prototype['defaultResponderId'] = undefined;

/**
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/DefaultResponderDefaultResponder} defaultResponder
 */
Whatsapp.prototype['defaultResponder'] = undefined;

// Implement Integration interface:
/**
 * The unique ID of the integration.
 * @member {String} id
 */
_Integration["default"].prototype['id'] = undefined;
/**
 * @member {String} type
 */
_Integration["default"].prototype['type'] = undefined;
/**
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/Status} status
 */
_Integration["default"].prototype['status'] = undefined;
/**
 * A human-friendly name used to identify the integration.
 * @member {String} displayName
 */
_Integration["default"].prototype['displayName'] = undefined;
// Implement WhatsappAllOf interface:
/**
 * To configure a WhatsApp integration, use your WhatsApp API Client connection information. Sunshine Conversations can provide WhatsApp API Client hosting for approved brands. See our [WhatsApp guide](https://docs.smooch.io/guide/whatsapp/#whatsapp-api-client) for more details on WhatsApp API Client hosting. 
 * @member {String} type
 * @default 'whatsapp'
 */
_WhatsappAllOf["default"].prototype['type'] = 'whatsapp';
/**
 * Specify a fallback language to use when sending WhatsApp message template using the short hand syntax. Defaults to en_US. See WhatsApp documentation for more info.
 * @member {String} hsmFallbackLanguage
 * @default 'en_US'
 */
_WhatsappAllOf["default"].prototype['hsmFallbackLanguage'] = 'en_US';
/**
 * The business ID associated with the WhatsApp account. In combination with accountManagementAccessToken, it’s used for Message Template Reconstruction.
 * @member {String} accountId
 */
_WhatsappAllOf["default"].prototype['accountId'] = undefined;
/**
 * An access token associated with the accountId used to query the WhatsApp Account Management API. In combination with accountId, it’s used for Message Template Reconstruction.
 * @member {String} accountManagementAccessToken
 */
_WhatsappAllOf["default"].prototype['accountManagementAccessToken'] = undefined;
/**
 * The phone number that is associated with the deployment of this integration, if one exists.
 * @member {String} phoneNumber
 */
_WhatsappAllOf["default"].prototype['phoneNumber'] = undefined;
// Implement DefaultResponderId interface:
/**
 * The default responder ID for the integration. This is the ID of the responder that will be used to send messages to the user. For more information, refer to the <a href=\"https://developer.zendesk.com/documentation/conversations/messaging-platform/programmable-conversations/switchboard/#default-integration-assignment\">Switchboard guide</a>. 
 * @member {String} defaultResponderId
 */
_DefaultResponderId["default"].prototype['defaultResponderId'] = undefined;
// Implement DefaultResponder interface:
/**
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/DefaultResponderDefaultResponder} defaultResponder
 */
_DefaultResponder["default"].prototype['defaultResponder'] = undefined;
var _default = exports["default"] = Whatsapp;
