// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.api/AppKeysApi.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _AppKeyCreateBody = _interopRequireDefault(require("../sunshine-conversations-client.model/AppKeyCreateBody"));
var _AppKeyListResponse = _interopRequireDefault(require("../sunshine-conversations-client.model/AppKeyListResponse"));
var _AppKeyResponse = _interopRequireDefault(require("../sunshine-conversations-client.model/AppKeyResponse"));
var _ErrorResponse = _interopRequireDefault(require("../sunshine-conversations-client.model/ErrorResponse"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(a, n) { if (!(a instanceof n)) throw new TypeError("Cannot call a class as a function"); }
function _defineProperties(e, r) { for (var t = 0; t < r.length; t++) { var o = r[t]; o.enumerable = o.enumerable || !1, o.configurable = !0, "value" in o && (o.writable = !0), Object.defineProperty(e, _toPropertyKey(o.key), o); } }
function _createClass(e, r, t) { return r && _defineProperties(e.prototype, r), t && _defineProperties(e, t), Object.defineProperty(e, "prototype", { writable: !1 }), e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Sunshine Conversations API
 *
 * The version of the OpenAPI document: 16.0.1
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 *
 * OpenAPI Generator version: 4.3.1
 *
 * Do not edit the class manually.
 *
 */
/**
* AppKeys service.
* @module sunshine-conversations-client/sunshine-conversations-client.api/AppKeysApi
* @version 16.0.1
*/
var AppKeysApi = exports["default"] = /*#__PURE__*/function () {
  /**
  * Constructs a new AppKeysApi. 
  * @alias module:sunshine-conversations-client/sunshine-conversations-client.api/AppKeysApi
  * @class
  * @param {module:sunshine-conversations-client/ApiClient} [apiClient] Optional API client implementation to use,
  * default to {@link module:sunshine-conversations-client/ApiClient#instance} if unspecified.
  */
  function AppKeysApi(apiClient) {
    _classCallCheck(this, AppKeysApi);
    this.apiClient = apiClient || _ApiClient["default"].instance;
  }

  /**
   * Create App Key
   * Creates an API key for the specified app. The response body will include a secret  as well as its corresponding id, which you can use to generate JSON Web Tokens to  securely make API calls on behalf of the app. 
   * @param {String} appId Identifies the app.
   * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/AppKeyCreateBody} appKeyCreateBody 
   * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with an object containing data of type {@link module:sunshine-conversations-client/sunshine-conversations-client.model/AppKeyResponse} and HTTP response
   */
  return _createClass(AppKeysApi, [{
    key: "createAppKeyWithHttpInfo",
    value: function createAppKeyWithHttpInfo(appId, appKeyCreateBody) {
      var postBody = appKeyCreateBody;
      // verify the required parameter 'appId' is set
      if (appId === undefined || appId === null) {
        throw new Error("Missing the required parameter 'appId' when calling createAppKey");
      }
      // verify the required parameter 'appKeyCreateBody' is set
      if (appKeyCreateBody === undefined || appKeyCreateBody === null) {
        throw new Error("Missing the required parameter 'appKeyCreateBody' when calling createAppKey");
      }
      var pathParams = {
        'appId': appId
      };
      var queryParams = {};
      var headerParams = {};
      var formParams = {};
      var authNames = ['basicAuth', 'bearerAuth'];
      var contentTypes = ['application/json'];
      var accepts = ['application/json'];
      var returnType = _AppKeyResponse["default"];
      return this.apiClient.callApi('/v2/apps/{appId}/keys', 'POST', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null);
    }

    /**
     * Create App Key
     * Creates an API key for the specified app. The response body will include a secret  as well as its corresponding id, which you can use to generate JSON Web Tokens to  securely make API calls on behalf of the app. 
     * @param {String} appId Identifies the app.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/AppKeyCreateBody} appKeyCreateBody 
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with data of type {@link module:sunshine-conversations-client/sunshine-conversations-client.model/AppKeyResponse}
     */
  }, {
    key: "createAppKey",
    value: function createAppKey(appId, appKeyCreateBody) {
      return this.createAppKeyWithHttpInfo(appId, appKeyCreateBody).then(function (response_and_data) {
        return response_and_data.data;
      });
    }

    /**
     * Delete App Key
     * Removes an API key.
     * @param {String} appId Identifies the app.
     * @param {String} keyId The id of the key.
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with an object containing data of type {@link Object} and HTTP response
     */
  }, {
    key: "deleteAppKeyWithHttpInfo",
    value: function deleteAppKeyWithHttpInfo(appId, keyId) {
      var postBody = null;
      // verify the required parameter 'appId' is set
      if (appId === undefined || appId === null) {
        throw new Error("Missing the required parameter 'appId' when calling deleteAppKey");
      }
      // verify the required parameter 'keyId' is set
      if (keyId === undefined || keyId === null) {
        throw new Error("Missing the required parameter 'keyId' when calling deleteAppKey");
      }
      var pathParams = {
        'appId': appId,
        'keyId': keyId
      };
      var queryParams = {};
      var headerParams = {};
      var formParams = {};
      var authNames = ['basicAuth', 'bearerAuth'];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = Object;
      return this.apiClient.callApi('/v2/apps/{appId}/keys/{keyId}', 'DELETE', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null);
    }

    /**
     * Delete App Key
     * Removes an API key.
     * @param {String} appId Identifies the app.
     * @param {String} keyId The id of the key.
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with data of type {@link Object}
     */
  }, {
    key: "deleteAppKey",
    value: function deleteAppKey(appId, keyId) {
      return this.deleteAppKeyWithHttpInfo(appId, keyId).then(function (response_and_data) {
        return response_and_data.data;
      });
    }

    /**
     * Get App Key
     * Returns an API key.
     * @param {String} appId Identifies the app.
     * @param {String} keyId The id of the key.
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with an object containing data of type {@link module:sunshine-conversations-client/sunshine-conversations-client.model/AppKeyResponse} and HTTP response
     */
  }, {
    key: "getAppKeyWithHttpInfo",
    value: function getAppKeyWithHttpInfo(appId, keyId) {
      var postBody = null;
      // verify the required parameter 'appId' is set
      if (appId === undefined || appId === null) {
        throw new Error("Missing the required parameter 'appId' when calling getAppKey");
      }
      // verify the required parameter 'keyId' is set
      if (keyId === undefined || keyId === null) {
        throw new Error("Missing the required parameter 'keyId' when calling getAppKey");
      }
      var pathParams = {
        'appId': appId,
        'keyId': keyId
      };
      var queryParams = {};
      var headerParams = {};
      var formParams = {};
      var authNames = ['basicAuth', 'bearerAuth'];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = _AppKeyResponse["default"];
      return this.apiClient.callApi('/v2/apps/{appId}/keys/{keyId}', 'GET', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null);
    }

    /**
     * Get App Key
     * Returns an API key.
     * @param {String} appId Identifies the app.
     * @param {String} keyId The id of the key.
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with data of type {@link module:sunshine-conversations-client/sunshine-conversations-client.model/AppKeyResponse}
     */
  }, {
    key: "getAppKey",
    value: function getAppKey(appId, keyId) {
      return this.getAppKeyWithHttpInfo(appId, keyId).then(function (response_and_data) {
        return response_and_data.data;
      });
    }

    /**
     * List App Keys
     * Lists all API keys for a given app.
     * @param {String} appId Identifies the app.
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with an object containing data of type {@link module:sunshine-conversations-client/sunshine-conversations-client.model/AppKeyListResponse} and HTTP response
     */
  }, {
    key: "listAppKeysWithHttpInfo",
    value: function listAppKeysWithHttpInfo(appId) {
      var postBody = null;
      // verify the required parameter 'appId' is set
      if (appId === undefined || appId === null) {
        throw new Error("Missing the required parameter 'appId' when calling listAppKeys");
      }
      var pathParams = {
        'appId': appId
      };
      var queryParams = {};
      var headerParams = {};
      var formParams = {};
      var authNames = ['basicAuth', 'bearerAuth'];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = _AppKeyListResponse["default"];
      return this.apiClient.callApi('/v2/apps/{appId}/keys', 'GET', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null);
    }

    /**
     * List App Keys
     * Lists all API keys for a given app.
     * @param {String} appId Identifies the app.
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with data of type {@link module:sunshine-conversations-client/sunshine-conversations-client.model/AppKeyListResponse}
     */
  }, {
    key: "listAppKeys",
    value: function listAppKeys(appId) {
      return this.listAppKeysWithHttpInfo(appId).then(function (response_and_data) {
        return response_and_data.data;
      });
    }
  }]);
}();
