// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.api/OAuthEndpointsApi.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _AppResponse = _interopRequireDefault(require("../sunshine-conversations-client.model/AppResponse"));
var _InlineObject = _interopRequireDefault(require("../sunshine-conversations-client.model/InlineObject"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(a, n) { if (!(a instanceof n)) throw new TypeError("Cannot call a class as a function"); }
function _defineProperties(e, r) { for (var t = 0; t < r.length; t++) { var o = r[t]; o.enumerable = o.enumerable || !1, o.configurable = !0, "value" in o && (o.writable = !0), Object.defineProperty(e, _toPropertyKey(o.key), o); } }
function _createClass(e, r, t) { return r && _defineProperties(e.prototype, r), t && _defineProperties(e, t), Object.defineProperty(e, "prototype", { writable: !1 }), e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Sunshine Conversations API
 *
 * The version of the OpenAPI document: 16.0.1
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 *
 * OpenAPI Generator version: 4.3.1
 *
 * Do not edit the class manually.
 *
 */
/**
* OAuthEndpoints service.
* @module sunshine-conversations-client/sunshine-conversations-client.api/OAuthEndpointsApi
* @version 16.0.1
*/
var OAuthEndpointsApi = exports["default"] = /*#__PURE__*/function () {
  /**
  * Constructs a new OAuthEndpointsApi. 
  * @alias module:sunshine-conversations-client/sunshine-conversations-client.api/OAuthEndpointsApi
  * @class
  * @param {module:sunshine-conversations-client/ApiClient} [apiClient] Optional API client implementation to use,
  * default to {@link module:sunshine-conversations-client/ApiClient#instance} if unspecified.
  */
  function OAuthEndpointsApi(apiClient) {
    _classCallCheck(this, OAuthEndpointsApi);
    this.apiClient = apiClient || _ApiClient["default"].instance;
  }

  /**
   * Authorize
   * This endpoint begins the OAuth flow. It relies on a browser session for authentication. If the user is not logged in to Zendesk they will be redirected to the login page.  If the user has many Zendesk accounts, they will first be prompted to select the one  they wish to integrate with. They will then be presented with an Allow/Deny dialog,  describing details of the access your integration is requesting.  Use `oauth-bridge.zendesk.com/sc` as the base URL when redirecting the user to this endpoint.  ``` https://oauth-bridge.zendesk.com/sc/oauth/authorize?response_type=code&client_id={client_id} ``` 
   * @param {String} clientId Your integration’s unique identifier
   * @param {String} responseType For now the only acceptable value is code.
   * @param {Object} opts Optional parameters
   * @param {String} opts.state You may pass in any arbitrary string value here which will be returned to you along with the code via browser redirect.
   * @param {String} opts.redirectUri You may pass in a redirect_uri to determine which URI the response is redirected to. This URI must be contained in the list configured by your integration. If this option is not passed, the first URI present in the list will be used.
   * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with an object containing HTTP response
   */
  return _createClass(OAuthEndpointsApi, [{
    key: "authorizeWithHttpInfo",
    value: function authorizeWithHttpInfo(clientId, responseType, opts) {
      opts = opts || {};
      var postBody = null;
      // verify the required parameter 'clientId' is set
      if (clientId === undefined || clientId === null) {
        throw new Error("Missing the required parameter 'clientId' when calling authorize");
      }
      // verify the required parameter 'responseType' is set
      if (responseType === undefined || responseType === null) {
        throw new Error("Missing the required parameter 'responseType' when calling authorize");
      }
      var pathParams = {};
      var queryParams = {
        'client_id': clientId,
        'response_type': responseType,
        'state': opts['state'],
        'redirect_uri': opts['redirectUri']
      };
      var headerParams = {};
      var formParams = {};
      var authNames = [];
      var contentTypes = [];
      var accepts = ['text/plain'];
      var returnType = null;
      return this.apiClient.callApi('/oauth/authorize', 'GET', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null);
    }

    /**
     * Authorize
     * This endpoint begins the OAuth flow. It relies on a browser session for authentication. If the user is not logged in to Zendesk they will be redirected to the login page.  If the user has many Zendesk accounts, they will first be prompted to select the one  they wish to integrate with. They will then be presented with an Allow/Deny dialog,  describing details of the access your integration is requesting.  Use `oauth-bridge.zendesk.com/sc` as the base URL when redirecting the user to this endpoint.  ``` https://oauth-bridge.zendesk.com/sc/oauth/authorize?response_type=code&client_id={client_id} ``` 
     * @param {String} clientId Your integration’s unique identifier
     * @param {String} responseType For now the only acceptable value is code.
     * @param {Object} opts Optional parameters
     * @param {String} opts.state You may pass in any arbitrary string value here which will be returned to you along with the code via browser redirect.
     * @param {String} opts.redirectUri You may pass in a redirect_uri to determine which URI the response is redirected to. This URI must be contained in the list configured by your integration. If this option is not passed, the first URI present in the list will be used.
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}
     */
  }, {
    key: "authorize",
    value: function authorize(clientId, responseType, opts) {
      return this.authorizeWithHttpInfo(clientId, responseType, opts).then(function (response_and_data) {
        return response_and_data.data;
      });
    }

    /**
     * Get Token
     * This endpoint is used to exchange an authorization code for an access token.  It should only be used in server-to-server calls.  Use `oauth-bridge.zendesk.com/sc` as the base URL when invoking this endpoint.  ``` POST https://oauth-bridge.zendesk.com/sc/oauth/token ``` 
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/InlineObject} inlineObject 
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with an object containing data of type {@link Object} and HTTP response
     */
  }, {
    key: "getTokenWithHttpInfo",
    value: function getTokenWithHttpInfo(inlineObject) {
      var postBody = inlineObject;
      // verify the required parameter 'inlineObject' is set
      if (inlineObject === undefined || inlineObject === null) {
        throw new Error("Missing the required parameter 'inlineObject' when calling getToken");
      }
      var pathParams = {};
      var queryParams = {};
      var headerParams = {};
      var formParams = {};
      var authNames = [];
      var contentTypes = ['application/json'];
      var accepts = ['application/json'];
      var returnType = Object;
      return this.apiClient.callApi('/oauth/token', 'POST', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null);
    }

    /**
     * Get Token
     * This endpoint is used to exchange an authorization code for an access token.  It should only be used in server-to-server calls.  Use `oauth-bridge.zendesk.com/sc` as the base URL when invoking this endpoint.  ``` POST https://oauth-bridge.zendesk.com/sc/oauth/token ``` 
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/InlineObject} inlineObject 
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with data of type {@link Object}
     */
  }, {
    key: "getToken",
    value: function getToken(inlineObject) {
      return this.getTokenWithHttpInfo(inlineObject).then(function (response_and_data) {
        return response_and_data.data;
      });
    }

    /**
     * Get Token Info
     * This endpoint can be used to retrieve the app linked to the OAuth token. Typically used after receiving an access token via OAuth, in order to retrieve the app's `id` and `subdomain` to be used in future calls. May also be used to confirm that the credentials are still valid.  Use `oauth-bridge.zendesk.com/sc` as the base URL when invoking this endpoint.  ``` GET https://oauth-bridge.zendesk.com/sc/v2/tokenInfo ``` 
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with an object containing data of type {@link module:sunshine-conversations-client/sunshine-conversations-client.model/AppResponse} and HTTP response
     */
  }, {
    key: "getTokenInfoWithHttpInfo",
    value: function getTokenInfoWithHttpInfo() {
      var postBody = null;
      var pathParams = {};
      var queryParams = {};
      var headerParams = {};
      var formParams = {};
      var authNames = [];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = _AppResponse["default"];
      return this.apiClient.callApi('/v2/tokenInfo', 'GET', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null);
    }

    /**
     * Get Token Info
     * This endpoint can be used to retrieve the app linked to the OAuth token. Typically used after receiving an access token via OAuth, in order to retrieve the app's `id` and `subdomain` to be used in future calls. May also be used to confirm that the credentials are still valid.  Use `oauth-bridge.zendesk.com/sc` as the base URL when invoking this endpoint.  ``` GET https://oauth-bridge.zendesk.com/sc/v2/tokenInfo ``` 
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with data of type {@link module:sunshine-conversations-client/sunshine-conversations-client.model/AppResponse}
     */
  }, {
    key: "getTokenInfo",
    value: function getTokenInfo() {
      return this.getTokenInfoWithHttpInfo().then(function (response_and_data) {
        return response_and_data.data;
      });
    }

    /**
     * Revoke Access
     * This endpoint is used to revoke your integration's access to the user's app. Revoking access means your integration will no longer be able to interact with the app, and any webhooks the integration had previously configured will be removed.  Calling this endpoint is equivalent to the user removing your integration manually. Your integration's `removeUrl` (if configured) will also be called when an integration is removed in this way. 
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with an object containing data of type {@link Object} and HTTP response
     */
  }, {
    key: "revokeAccessWithHttpInfo",
    value: function revokeAccessWithHttpInfo() {
      var postBody = null;
      var pathParams = {};
      var queryParams = {};
      var headerParams = {};
      var formParams = {};
      var authNames = ['basicAuth', 'bearerAuth'];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = Object;
      return this.apiClient.callApi('/oauth/authorization', 'DELETE', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null);
    }

    /**
     * Revoke Access
     * This endpoint is used to revoke your integration's access to the user's app. Revoking access means your integration will no longer be able to interact with the app, and any webhooks the integration had previously configured will be removed.  Calling this endpoint is equivalent to the user removing your integration manually. Your integration's `removeUrl` (if configured) will also be called when an integration is removed in this way. 
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with data of type {@link Object}
     */
  }, {
    key: "revokeAccess",
    value: function revokeAccess() {
      return this.revokeAccessWithHttpInfo().then(function (response_and_data) {
        return response_and_data.data;
      });
    }
  }]);
}();
